// webuikit/es/helpers/story-stack.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Wrapper = exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _spacing = _interopRequireDefault(require("../tokens/spacing"));

var alignItems = {
  top: "flex-start",
  center: "center",
  bottom: "flex-end"
};
var justifyContent = {
  fill: "space-between",
  "space-between": "space-between",
  left: "flex-start",
  right: "flex-end"
};

var StyledStack = _styledComponents["default"].div.withConfig({
  displayName: "story-stack__StyledStack",
  componentId: "avncrh-0"
})(["display:flex;flex-direction:row;align-items:", ";justify-content:", ";line-height:inherit;> *{flex:", ";margin-right:", ";}> *:last-child{margin-right:0;}"], function (props) {
  return alignItems[props.alignVertical];
}, function (props) {
  return justifyContent[props.align];
}, function (props) {
  return props.align === "fill" ? 1 : "none";
}, function (props) {
  return props.align === "fill" ? _spacing["default"].xsmall : 0;
});

var StackedItem = _styledComponents["default"].div.withConfig({
  displayName: "story-stack__StackedItem",
  componentId: "avncrh-1"
})(["flex-basis:", "%;line-height:inherit;"], function (props) {
  return props.width;
});

var Stack = function Stack(props) {
  var children;

  if (props.align === "fill" || props.align === "space-between") {
    children = _react["default"].Children.map(props.children, function (child, index) {
      var width = 0;
      if (props.widths) width = "".concat(props.widths[index]) || 0;
      return _react["default"].createElement(StackedItem, {
        width: width
      }, child);
    });
  } else {
    children = props.children;
  }

  return _react["default"].createElement(StyledStack, (0, _extends2["default"])({}, props, {
    align: props.align
  }), children);
};

Stack.propTypes = {
  /** Use align for stacking elements without margin between them */
  align: _propTypes["default"].oneOf(["fill", "left", "right", "space-between"]),

  /** Vertically align */
  alignVertical: _propTypes["default"].oneOf(["top", "center", "bottom"]),
  widths: _propTypes["default"].array,
  children: _propTypes["default"].Children
};
Stack.defaultProps = {
  align: "fill",
  alignVertical: "center"
};

var Wrapper = _styledComponents["default"].div.withConfig({
  displayName: "story-stack__Wrapper",
  componentId: "avncrh-2"
})(["display:flex;justify-content:space-between;flex-wrap:wrap;& > *{margin-right:20px;margin-bottom:20px;}"]);

exports.Wrapper = Wrapper;
var _default = Stack;
exports["default"] = _default;
