// webuikit/esm/Logo/Store.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _commonHelper = require("../helpers/commonHelper");

var WIDTH_THRESHOLD = 40; // Actual image sizes

var DEFAULT_WIDTH = 137;
var WIDTH_2X = 274;
var WIDTH_3X = 411;
var logo_urls = {
  playstore: {
    url_logo_1x: "data/webuikit/61f7c9690d8ba4525ba863d0914ef2aa1556001313.png",
    url_logo_2x: "data/webuikit/23e930757c3df49840c482a8638bf5c31556001144.png",
    url_logo_3x: "data/webuikit/171050f49cdf259cff90fa3b21a87ccd1556001348.png",
    url_logo_4x: "data/webuikit/a336792fc43e33ed4df43cfd3ff542431556001376.png"
  },
  appstore: {
    url_logo_1x: "data/webuikit/69c8ac2b881b7ae167c36b480580274e1556003577.png",
    url_logo_2x: "data/webuikit/9f0c85a5e33adb783fa0aef667075f9e1556003622.png",
    url_logo_3x: "data/webuikit/bd19aa0be8dd3a92e58861cb82b9bccb1556003662.png",
    url_logo_4x: "data/webuikit/9b91a3427bae3d2bf001e3a97f52c7111556003706.png"
  }
};

var Store = function Store(props) {
  var _props$type = props.type,
      type = _props$type === void 0 ? "appstore" : _props$type,
      _props$width = props.width,
      width = _props$width === void 0 ? DEFAULT_WIDTH : _props$width,
      _props$link = props.link,
      link = _props$link === void 0 ? "" : _props$link;
  var height = parseInt(width * (40 / 137));
  var url = getUrl(type, width);
  return _react["default"].createElement(Store.A, {
    href: link,
    "aria-label": "app store link"
  }, _react["default"].createElement(Store.Image, {
    width: width,
    height: height,
    url: url
  }));
};

var getUrl = function getUrl(type, width) {
  if (width <= DEFAULT_WIDTH + WIDTH_THRESHOLD) {
    return (0, _commonHelper.getCDN)(logo_urls[type].url_logo_1x);
  }

  if (width <= WIDTH_2X + WIDTH_THRESHOLD) {
    return (0, _commonHelper.getCDN)(logo_urls[type].url_logo_2x);
  }

  if (width <= WIDTH_3X + WIDTH_THRESHOLD) {
    return (0, _commonHelper.getCDN)(logo_urls[type].url_logo_3x);
  }

  return (0, _commonHelper.getCDN)(logo_urls[type].url_logo_4x);
};

Store.A = _styledComponents["default"].a.withConfig({
  displayName: "Store__A",
  componentId: "oze4g0-0"
})(["text-decoration:none;"]);
Store.Image = _styledComponents["default"].div.withConfig({
  displayName: "Store__Image",
  componentId: "oze4g0-1"
})(["width:", "rem;height:", "rem;background-image:url(", ");background-repeat:no-repeat;background-size:100%;@media (max-width:480px){width:11rem;height:calc(11rem * (40 / 137));}"], function (props) {
  return props.width / 10;
}, function (props) {
  return props.height / 10;
}, function (props) {
  return props.url;
});
Store.propTypes = {
  type: _propTypes["default"].oneOf(["appstore", "playstore"]),
  width: _propTypes["default"].number,
  link: _propTypes["default"].string
};
var _default = Store;
exports["default"] = _default;
