// webuikit/tokens/color/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "black", {
  enumerable: true,
  get: function get() {
    return _black["default"];
  }
});
Object.defineProperty(exports, "grey", {
  enumerable: true,
  get: function get() {
    return _grey["default"];
  }
});
Object.defineProperty(exports, "brown", {
  enumerable: true,
  get: function get() {
    return _brown["default"];
  }
});
Object.defineProperty(exports, "green", {
  enumerable: true,
  get: function get() {
    return _green["default"];
  }
});
Object.defineProperty(exports, "indigo", {
  enumerable: true,
  get: function get() {
    return _indigo["default"];
  }
});
Object.defineProperty(exports, "orange", {
  enumerable: true,
  get: function get() {
    return _orange["default"];
  }
});
Object.defineProperty(exports, "pink", {
  enumerable: true,
  get: function get() {
    return _pink["default"];
  }
});
Object.defineProperty(exports, "purple", {
  enumerable: true,
  get: function get() {
    return _purple["default"];
  }
});
Object.defineProperty(exports, "red", {
  enumerable: true,
  get: function get() {
    return _red["default"];
  }
});
Object.defineProperty(exports, "teal", {
  enumerable: true,
  get: function get() {
    return _teal["default"];
  }
});
Object.defineProperty(exports, "white", {
  enumerable: true,
  get: function get() {
    return _white["default"];
  }
});
Object.defineProperty(exports, "yellow", {
  enumerable: true,
  get: function get() {
    return _yellow["default"];
  }
});
Object.defineProperty(exports, "blue", {
  enumerable: true,
  get: function get() {
    return _blue["default"];
  }
});

var _black = _interopRequireDefault(require("./black"));

var _grey = _interopRequireDefault(require("./grey"));

var _brown = _interopRequireDefault(require("./brown"));

var _green = _interopRequireDefault(require("./green"));

var _indigo = _interopRequireDefault(require("./indigo"));

var _orange = _interopRequireDefault(require("./orange"));

var _pink = _interopRequireDefault(require("./pink"));

var _purple = _interopRequireDefault(require("./purple"));

var _red = _interopRequireDefault(require("./red"));

var _teal = _interopRequireDefault(require("./teal"));

var _white = _interopRequireDefault(require("./white"));

var _yellow = _interopRequireDefault(require("./yellow"));

var _blue = _interopRequireDefault(require("./blue"));
