// webuikit/es/Carousel/Carousel.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _lodashEs = require("lodash-es");

var _reactSlick = _interopRequireDefault(require("react-slick"));

require("style-loader!slick-carousel/slick/slick.css");

require("style-loader!slick-carousel/slick/slick-theme.css");

/* eslint-disable import/no-unresolved */
// libraries
// components
// styles
var Carousel = function Carousel(props) {
  var componentData = props.componentData,
      componentToRender = props.componentToRender;
  var DesiredComp = componentToRender;
  var settings = {
    dots: props.dots,
    infinite: props.infinite,
    speed: props.speed,
    slidesToShow: props.slidesToShow,
    slidesToScroll: props.slidesToScroll
  };
  return _react["default"].createElement(_reactSlick["default"].Element, (0, _extends2["default"])({}, settings, {
    showComponents: props.showComponents,
    carouselWidth: props.carouselWidth
  }), componentData.map(function (data, index) {
    return _react["default"].createElement(DesiredCompWrapper, {
      className: "carousel-content",
      key: index,
      height: props.height
    }, _react["default"].createElement(DesiredComp, data));
  }));
};

var DesiredCompWrapper = _styledComponents["default"].section.withConfig({
  displayName: "Carousel__DesiredCompWrapper",
  componentId: "arq64n-0"
})(["overflow:hidden;height:", ";"], function (props) {
  return props.height;
});

_reactSlick["default"].Element = (0, _styledComponents["default"])(_reactSlick["default"]).withConfig({
  displayName: "Carousel__Element",
  componentId: "arq64n-1"
})(["touch-action:manipulation;width:", ";.text-size{font-size:1.4rem;}.slick-track{display:flex;}.slick-next:before{top:1.3rem;left:1.1rem;transform:rotate(-45deg);}.slick-prev:before{top:1.3rem;left:1.4rem;transform:rotate(135deg);}.slick-prev{left:-1.7rem;}.slick-next{right:-1.7rem;}.slick-arrow{font-size:0;height:3.4rem;width:3.4rem;background:#ffffff;box-shadow:-4px 9px 21px rgba(60,64,69,0.2);border-radius:50%;position:absolute;z-index:2;&:hover,&:focus{background:#fdfdfd;}}.slick-prev:before,.slick-next:before{position:absolute;width:0.9rem;height:0.9rem;content:\"\";border-color:#000;border-style:solid;border-top:0;border-right-width:2px;border-bottom-width:2px;border-left:0;font-family:slick;font-size:13.5px;color:#696969;font-weight:bold;}.slick-slide{background:white;}.slick-dots{display:flex !important;justify-content:center;font-size:22px;li.slick-active::before{color:#5656ff;}li{margin:0;}li button:before{font-size:11px;}}"], function (props) {
  return props.carouselWidth;
});
Carousel.propTypes = {
  componentData: _propTypes["default"].array.isRequired,
  componentToRender: _propTypes["default"].node.isRequired,
  height: _propTypes["default"].string,
  carouselWidth: _propTypes["default"].string,
  dots: _propTypes["default"].bool,
  infinite: _propTypes["default"].bool,
  speed: _propTypes["default"].number,
  slidesToShow: _propTypes["default"].number,
  slidesToScroll: _propTypes["default"].number,
  showComponents: _propTypes["default"].bool
};
Carousel.defaultProps = {
  height: "200px",
  carouselWidth: "100%",
  dots: true,
  infinite: true,
  speed: 500,
  slidesToShow: 1,
  slidesToScroll: 1
};
var _default = Carousel;
exports["default"] = _default;
