// webuikit/esm/tokens/sizes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.imgUploaderDefault = exports.imgUploaderError = exports.radioViewDisplay = exports.inputPadding = exports.inputPhDefault = exports.inputPhFocused = exports.input = exports.button = void 0;
var button = {
  large: {
    height: "48px"
  },
  "default": {
    height: "44px"
  },
  compressed: {
    height: "36px"
  },
  small: {
    height: "32px"
  }
};
exports.button = button;
var input = {
  fontSize: "18px",
  width: "300px",
  height: "48px"
};
exports.input = input;
var inputPhFocused = {
  top: "-5px",
  left: "10px",
  fontSize: "11px",
  padding: "0 5px"
};
exports.inputPhFocused = inputPhFocused;
var inputPhDefault = {
  top: "15px",
  left: "12px",
  fontSize: "18px",
  padding: "0"
};
exports.inputPhDefault = inputPhDefault;
var inputPadding = {
  vPadding: 1.3,
  hzPadding: 1.2
};
exports.inputPadding = inputPadding;
var imgUploaderError = {
  tagFontSize: 20,
  tagLineHeight: 28,
  imgHeight: 34
};
exports.imgUploaderError = imgUploaderError;
var imgUploaderDefault = {
  tagFontSize: 15,
  tagLineHeight: 20,
  imgHeight: 43
};
exports.imgUploaderDefault = imgUploaderDefault;
var radioViewDisplay = {
  horizontal: {
    display: "flex"
  },
  "default": {
    display: "block"
  }
};
exports.radioViewDisplay = radioViewDisplay;
