// webuikit/esm/ImageUploader/componentHelpers.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.viewRenderer = exports.eventManager = void 0;

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _react = _interopRequireDefault(require("react"));

var _lodashEs = require("lodash-es");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _imageUploaderHelpers = require("./imageUploaderHelpers");

// Libraries
// helpers
var sendData = function sendData(_ref) {
  var onChange = _ref.onChange,
      setUrlMap = _ref.setUrlMap,
      _ref$urlMap = _ref.urlMap,
      urlMap = _ref$urlMap === void 0 ? [] : _ref$urlMap;
  return function (_ref2) {
    var _ref2$target = _ref2.target,
        target = _ref2$target === void 0 ? {} : _ref2$target;
    var files = target.files;
    var URL = window.URL || window.webkitURL;
    var newUrlMap = (0, _toConsumableArray2["default"])(urlMap);
    (0, _lodashEs.forEach)(files, function (file) {
      newUrlMap.push(URL.createObjectURL(file));
    });
    setUrlMap(newUrlMap);
    var isValid = onChange(files);
    return isValid;
  };
};

var handleDrop = function handleDrop(params) {
  return function (e) {
    e.preventDefault();
    e.stopPropagation();
    var uploadObj = {
      target: e.nativeEvent.dataTransfer
    };
    params.setHover(false);
    sendData(params)(uploadObj);
  };
};

var handleDragEnter = function handleDragEnter(setHover) {
  return function (e) {
    e.preventDefault();
    e.stopPropagation();
    setHover(true);
  };
}; // eslint-disable-next-line no-unused-vars


var handleDragLeave = function handleDragLeave(setHover) {
  return function (e) {
    setHover(false);
  };
};

var handleDragOver = function handleDragOver(e) {
  e.preventDefault();
};

var eventManager = function eventManager(func, params) {
  func("drop", handleDrop(params));
};

exports.eventManager = eventManager;

var inputRenderer = function inputRenderer(params) {
  return _react["default"].createElement("input", {
    type: "file",
    name: "myFile",
    accept: "image/*",
    onChange: sendData(params),
    multiple: params.multiple
  });
};

var dropTargetRenderer = function dropTargetRenderer(params) {
  return _react["default"].createElement("div", {
    className: "drop-target",
    onDrop: handleDrop(params),
    onDragEnter: handleDragEnter(params.setHover),
    onDragOver: handleDragOver,
    onDragLeave: handleDragLeave(params.setHover)
  });
};

var urlView = function urlView(params) {
  return _react["default"].createElement(ImgWrapper, null, params.urlMap.map(function (url, i) {
    return _react["default"].createElement("div", {
      key: i
    }, _react["default"].createElement(Img, {
      src: url,
      error: params.errMap[i]
    }), params.errMap[i] && _react["default"].createElement(Error, null, params.errMap[i]));
  }), _react["default"].createElement(BtnHolder, {
    image: true
  }, _react["default"].createElement(Img, {
    src: "https://b.zmtcdn.com/data/webuikit/29cea710c011251e318601171b565fa71556795839.png",
    logo: true
  }), inputRenderer(params)));
};

var viewRenderer = function viewRenderer(params) {
  return _react["default"].createElement(Article, {
    hover: params.hover
  }, dropTargetRenderer(params), _react["default"].createElement(Section, null, _react["default"].createElement("div", {
    className: "btn-wrapper-url-view"
  }, "Drag & Drop to upload or", _react["default"].createElement(BtnHolder, null, " Browse", inputRenderer(params))), urlView(params)));
};

exports.viewRenderer = viewRenderer;

var ImgWrapper = _styledComponents["default"].section.withConfig({
  displayName: "componentHelpers__ImgWrapper",
  componentId: "sc-59rmw3-0"
})(["margin:0 0.5rem 3.2rem;display:flex;flex-wrap:wrap;justify-content:center;"]);

var Error = _styledComponents["default"].div.withConfig({
  displayName: "componentHelpers__Error",
  componentId: "sc-59rmw3-1"
})(["font-size:1.2rem;color:red;"]);

var Img = _styledComponents["default"].img.withConfig({
  displayName: "componentHelpers__Img",
  componentId: "sc-59rmw3-2"
})(["width:10rem;height:10rem;border-radius:4px;margin-right:1.6rem;border:", ";"], _imageUploaderHelpers.getImageBorder);

var Section = _styledComponents["default"].section.withConfig({
  displayName: "componentHelpers__Section",
  componentId: "sc-59rmw3-3"
})([".btn-wrapper-url-view{display:flex;justify-content:center;padding:3rem 0 3.5rem;color:#9c9c9c;}input{cursor:pointer;}.btn-tag{color:", ";font-size:", "px;line-height:", "px;}"], (0, _imageUploaderHelpers.getAttributes)("tagColor"), (0, _imageUploaderHelpers.getAttributes)("tagFontSize"), (0, _imageUploaderHelpers.getAttributes)("tagLineHeight"));

var BtnHolder = _styledComponents["default"].span.withConfig({
  displayName: "componentHelpers__BtnHolder",
  componentId: "sc-59rmw3-4"
})(["position:relative;overflow:hidden;display:inline-block;color:#363636;padding-left:0.5rem;text-decoration:underline;input[type=\"file\"]{font-size:", ";position:absolute;left:0;top:0;opacity:0;}"], function (props) {
  return props.image ? "10rem" : "1rem";
});

var Article = _styledComponents["default"].article.withConfig({
  displayName: "componentHelpers__Article",
  componentId: "sc-59rmw3-5"
})(["border:1.5px dotted ", ";background:", ";border-radius:2px;position:relative;opacity:", ";width:100%;.logo{height:", "px;}.drop-target{position:absolute;left:0;top:0;width:100%;height:100%;}"], (0, _imageUploaderHelpers.getAttributes)("borderColor"), (0, _imageUploaderHelpers.getAttributes)("background"), _imageUploaderHelpers.getOpacity, (0, _imageUploaderHelpers.getAttributes)("imgHeight"));
