// webuikit/es/Layouts/PageLayout.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Gap2 = exports.Gap10 = exports.Gap = exports.CenteredP = exports.CenteredH1 = exports.CenteredH2 = exports.StackedSectionLeft = exports.StackedSection = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _color = require("../tokens/color");

var _Typography = require("../Typography");

var Main = function Main(props) {
  return _react["default"].createElement(Main.Wrapper, null, _react["default"].createElement(Main.content, null, props.children));
};

Main.Wrapper = _styledComponents["default"].main.withConfig({
  displayName: "PageLayout__Wrapper",
  componentId: "py6dyo-0"
})(["width:100%;max-width:100%;justify-content:center;align-items:center;background-color:", ";display:flex;justify-content:center;"], _color.white);
Main.content = _styledComponents["default"].div.withConfig({
  displayName: "PageLayout__content",
  componentId: "py6dyo-1"
})(["width:100%;max-width:90rem;margin:0 auto;display:flex;flex-wrap:wrap;height:auto;align-items:center;flex-direction:column;"]);

var StackedSection = _styledComponents["default"].section.withConfig({
  displayName: "PageLayout__StackedSection",
  componentId: "py6dyo-2"
})(["display:flex;flex-direction:column;align-items:center;"]);

exports.StackedSection = StackedSection;
var StackedSectionLeft = (0, _styledComponents["default"])(StackedSection).withConfig({
  displayName: "PageLayout__StackedSectionLeft",
  componentId: "py6dyo-3"
})(["align-items:start;max-width:", ";width:100%;align-self:flex-start;"], function (props) {
  return props.maxWidth || "90rem";
});
exports.StackedSectionLeft = StackedSectionLeft;
var CenteredH2 = (0, _styledComponents["default"])(_Typography.H2).withConfig({
  displayName: "PageLayout__CenteredH2",
  componentId: "py6dyo-4"
})(["width:100%;text-align:center;"]);
exports.CenteredH2 = CenteredH2;
var CenteredH1 = (0, _styledComponents["default"])(_Typography.H1).withConfig({
  displayName: "PageLayout__CenteredH1",
  componentId: "py6dyo-5"
})(["width:100%;text-align:center;"]);
exports.CenteredH1 = CenteredH1;
var CenteredP = (0, _styledComponents["default"])(_Typography.P).withConfig({
  displayName: "PageLayout__CenteredP",
  componentId: "py6dyo-6"
})(["width:80%;text-align:center;margin:0 auto;"]);
exports.CenteredP = CenteredP;

var Gap = _styledComponents["default"].div.withConfig({
  displayName: "PageLayout__Gap",
  componentId: "py6dyo-7"
})(["margin-bottom:14rem;"]);

exports.Gap = Gap;

var Gap10 = _styledComponents["default"].div.withConfig({
  displayName: "PageLayout__Gap10",
  componentId: "py6dyo-8"
})(["margin-bottom:10rem;"]);

exports.Gap10 = Gap10;

var Gap2 = _styledComponents["default"].div.withConfig({
  displayName: "PageLayout__Gap2",
  componentId: "py6dyo-9"
})(["margin-top:23rem;"]);

exports.Gap2 = Gap2;
var _default = Main;
exports["default"] = _default;
