// webuikit/es/Footer/Footer.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _color = require("../tokens/color");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Icon = _interopRequireDefault(require("../Icon"));

var _Logo = require("../Logo");

var _CountryDropdown = _interopRequireDefault(require("../CountryDropdown"));

var _LanguageDropdown = _interopRequireDefault(require("../LanguageDropdown"));

var _lodashEs = require("lodash-es");

var _contents = require("./contents");

var Footer = function Footer(props) {
  var _props$onCountrySelec = props.onCountrySelect,
      onCountrySelect = _props$onCountrySelec === void 0 ? _lodashEs.noop : _props$onCountrySelec,
      _props$onLanguageSele = props.onLanguageSelect,
      onLanguageSelect = _props$onLanguageSele === void 0 ? _lodashEs.noop : _props$onLanguageSele;
  return _react["default"].createElement(Footer.Wrapper, null, _react["default"].createElement(Footer.Content, null, _react["default"].createElement(Footer.List, null, _react["default"].createElement(Footer.Locale, {
    onCountrySelect: onCountrySelect,
    onLanguageSelect: onLanguageSelect
  }), _react["default"].createElement(Footer.Column, {
    name: "Company",
    content: _contents.companyContent
  }), _react["default"].createElement(Footer.Column, {
    name: "For Foodies",
    content: _contents.foodiesContent
  }), _react["default"].createElement(Footer.Column, {
    name: "For Restaurants",
    content: _contents.restaurantsContent
  }), _react["default"].createElement(Footer.Column, {
    name: "For you",
    content: _contents.forYouContent
  }), _react["default"].createElement(Footer.Links, {
    name: "Social links",
    icons: _contents.icons
  })), _react["default"].createElement(Footer.HR, null), _react["default"].createElement(Footer.Disclaimer, null, _contents.disclaimer)));
};

Footer.propTypes = {
  onCountrySelect: _propTypes["default"].func,
  onLanguageSelect: _propTypes["default"].func
};
Footer.Disclaimer = _styledComponents["default"].p.withConfig({
  displayName: "Footer__Disclaimer",
  componentId: "sc-12ft0j5-0"
})(["font-size:1.1rem;color:", ";text-align:center;line-height:1.3rem;margin-top:2.3rem;@media (max-width:480px){margin-bottom:3rem;}"], _color.grey.z800);

Footer.Links = function (props) {
  var _props$name = props.name,
      name = _props$name === void 0 ? "" : _props$name,
      _props$icons = props.icons,
      icons = _props$icons === void 0 ? [] : _props$icons;
  return _react["default"].createElement(Footer.LinksContainer, null, _react["default"].createElement(Footer.Heading, null, name), _react["default"].createElement(Footer.IconContainer, null, icons.map(function (icon) {
    return _react["default"].createElement(Footer.Item.A, {
      key: icon.name,
      href: icon.link,
      isIcon: true
    }, _react["default"].createElement(_Icon["default"], {
      name: icon.name,
      size: 20
    }));
  })));
};

Footer.Column = function (props) {
  var _props$name2 = props.name,
      name = _props$name2 === void 0 ? "" : _props$name2,
      _props$content = props.content,
      content = _props$content === void 0 ? [] : _props$content;
  return _react["default"].createElement(Footer.ColumnContainer, null, _react["default"].createElement(Footer.Heading, null, name), _react["default"].createElement("nav", null, content.map(function (row) {
    return _react["default"].createElement(Footer.Item, {
      key: row.label,
      label: row.label,
      link: row.label
    });
  })));
};

Footer.Item = function (props) {
  return _react["default"].createElement(Footer.Item.A, {
    href: props.link
  }, _react["default"].createElement(Footer.Item.P, null, props.label));
};

Footer.Locale = function (props) {
  var _props$onCountrySelec2 = props.onCountrySelect,
      onCountrySelect = _props$onCountrySelec2 === void 0 ? _lodashEs.noop : _props$onCountrySelec2,
      _props$onLanguageSele2 = props.onLanguageSelect,
      onLanguageSelect = _props$onLanguageSele2 === void 0 ? _lodashEs.noop : _props$onLanguageSele2;

  var handleCountrySelect = function handleCountrySelect(country) {
    return onCountrySelect(country);
  };

  var handleLangSelect = function handleLangSelect(lang) {
    return onLanguageSelect(lang);
  };

  return _react["default"].createElement(Footer.LocaleContainer, null, _react["default"].createElement(Footer.DropdownContainer, null, _react["default"].createElement(_CountryDropdown["default"], {
    country: "India",
    value: "india",
    countries: _contents.countries,
    onSelect: handleCountrySelect
  }), _react["default"].createElement(Gap, null), _react["default"].createElement(_LanguageDropdown["default"], {
    lang: "English",
    value: "EN",
    languages: _contents.languages,
    onSelect: handleLangSelect
  })), _react["default"].createElement(Gap, null), _react["default"].createElement(Footer.StoreLinksContainer, null, _react["default"].createElement(_Logo.Appstore, {
    link: "/appstore"
  }), _react["default"].createElement(Gap, null), _react["default"].createElement(_Logo.Playstore, {
    link: "/playstore"
  })));
};

Footer.Locale.propTypes = {
  onCountrySelect: _propTypes["default"].func,
  onLanguageSelect: _propTypes["default"].func
};
Footer.DropdownContainer = _styledComponents["default"].section.withConfig({
  displayName: "Footer__DropdownContainer",
  componentId: "sc-12ft0j5-1"
})(["@media (max-width:768px) and (min-width:481px){display:flex;align-items:center;}"]);
Footer.StoreLinksContainer = _styledComponents["default"].section.withConfig({
  displayName: "Footer__StoreLinksContainer",
  componentId: "sc-12ft0j5-2"
})(["@media (max-width:768px) and (min-width:481px){display:flex;align-items:center;}"]);

var Gap = _styledComponents["default"].div.withConfig({
  displayName: "Footer__Gap",
  componentId: "sc-12ft0j5-3"
})(["margin-top:1.2rem;@media (max-width:768px) and (min-width:481px){margin-right:1.2rem;}"]);

Footer.Links.propTypes = {
  name: _propTypes["default"].string,
  icons: _propTypes["default"].array
};
Footer.Column.propTypes = {
  name: _propTypes["default"].string,
  content: _propTypes["default"].array
};
Footer.Item.propTypes = {
  label: _propTypes["default"].string,
  link: _propTypes["default"].string
};
Footer.HR = _styledComponents["default"].hr.withConfig({
  displayName: "Footer__HR",
  componentId: "sc-12ft0j5-4"
})(["margin-top:5rem;border:0;border-top:1px solid ", ";@media (max-width:480px){margin-top:-0.6rem;}"], _color.grey.z300);
Footer.LinksContainer = _styledComponents["default"].div.withConfig({
  displayName: "Footer__LinksContainer",
  componentId: "sc-12ft0j5-5"
})(["display:flex;flex-direction:column;"]);
Footer.IconContainer = _styledComponents["default"].div.withConfig({
  displayName: "Footer__IconContainer",
  componentId: "sc-12ft0j5-6"
})(["display:flex;justify-content:flex-start;@media (max-width:550px) and (min-width:481px){flex-direction:column;}"]);
Footer.Item.P = _styledComponents["default"].p.withConfig({
  displayName: "Footer__P",
  componentId: "sc-12ft0j5-7"
})(["font-size:1.4rem;line-height:2.1rem;color:", ";margin:0.5rem 0;@media (max-width:480px){font-size:1.3rem;line-height:1.6rem;}"], _color.grey.z800);
Footer.Item.A = _styledComponents["default"].a.withConfig({
  displayName: "Footer__A",
  componentId: "sc-12ft0j5-8"
})(["text-decoration:none;", ""], function (props) {
  return props.isIcon && (0, _styledComponents.css)(["margin-right:0.6rem;@media (max-width:550px) and (min-width:481px){margin-top:0.6rem;}"]);
});
Footer.Heading = _styledComponents["default"].h6.withConfig({
  displayName: "Footer__Heading",
  componentId: "sc-12ft0j5-9"
})(["font-size:1.8rem;font-weight:500;margin:0 0 1.2rem 0;"]);
Footer.LocaleContainer = _styledComponents["default"].div.withConfig({
  displayName: "Footer__LocaleContainer",
  componentId: "sc-12ft0j5-10"
})(["width:20rem;@media (max-width:480px){width:unset;margin-bottom:3rem;}@media (max-width:768px) and (min-width:481px){position:absolute;top:4rem;display:flex;align-items:center;justify-content:space-between;width:calc(100% - 4.8rem);}"]);
Footer.ColumnContainer = _styledComponents["default"].div.withConfig({
  displayName: "Footer__ColumnContainer",
  componentId: "sc-12ft0j5-11"
})(["margin-right:1rem;width:max-content;@media (max-width:480px){margin-bottom:3rem;}"]);
Footer.Wrapper = _styledComponents["default"].footer.withConfig({
  displayName: "Footer__Wrapper",
  componentId: "sc-12ft0j5-12"
})(["width:100%;background:", ";display:flex;justify-content:center;margin-top:5rem;"], _color.grey.z100);
Footer.Content = _styledComponents["default"].div.withConfig({
  displayName: "Footer__Content",
  componentId: "sc-12ft0j5-13"
})(["max-width:90rem;width:90rem;padding:4.8rem 0 2.2rem 0;position:relative;@media (max-width:768px) and (min-width:481px){padding:4.8rem 2.4rem 2.2rem 2.4rem;}@media (max-width:480px){padding:4.8rem 2.4rem 2.2rem 2.4rem;}"]);
Footer.List = _styledComponents["default"].section.withConfig({
  displayName: "Footer__List",
  componentId: "sc-12ft0j5-14"
})(["display flex;justify-content:space-between;@media (max-width:480px){display:grid;grid-template-areas:\"first second\";}@media (max-width:768px) and (min-width:481px){margin-top:10.5rem;}"]);
var _default = Footer;
exports["default"] = _default;
