// webuikit/esm/Typography/Heading.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.H6 = exports.H5 = exports.H4 = exports.H3 = exports.H2 = exports.H1 = void 0;

var _typography = require("../tokens/typography");

var _color = require("../tokens/color");

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var defaultMargin = (0, _styledComponents.css)(["margin-block-start:1rem;margin-block-end:1rem;"]);

var H1 = _styledComponents["default"].h1.withConfig({
  displayName: "Heading__H1",
  componentId: "xu06vv-0"
})(["font-size:", ";line-height:", ";font-weight:", ";color:", ";", ""], function (props) {
  return props.fontSize || _typography.fontSizes.z900;
}, _typography.lineHeight.regular, _typography.fontWeights.regular, function (props) {
  return props.color || _color.black;
}, defaultMargin);

exports.H1 = H1;

var H2 = _styledComponents["default"].h2.withConfig({
  displayName: "Heading__H2",
  componentId: "xu06vv-1"
})(["font-size:1.5rem;font-size:", ";line-height:", ";font-weight:", ";color:", ";", ""], function (props) {
  return props.fontSize || _typography.fontSizes.z800;
}, _typography.lineHeight.regular, _typography.fontWeights.regular, function (props) {
  return props.color || _color.black;
}, defaultMargin);

exports.H2 = H2;

var H3 = _styledComponents["default"].h3.withConfig({
  displayName: "Heading__H3",
  componentId: "xu06vv-2"
})(["font-size:", ";line-height:", ";font-weight:", ";color:", ";", ""], function (props) {
  return props.fontSize || _typography.fontSizes.z700;
}, _typography.lineHeight.regular, _typography.fontWeights.regular, function (props) {
  return props.color || _color.black;
}, defaultMargin);

exports.H3 = H3;

var H4 = _styledComponents["default"].h4.withConfig({
  displayName: "Heading__H4",
  componentId: "xu06vv-3"
})(["font-size:", ";line-height:", ";font-weight:", ";color:", ";", " margin-block-end:0;"], function (props) {
  return props.fontSize || _typography.fontSizes.z600;
}, _typography.lineHeight.regular, _typography.fontWeights.regular, function (props) {
  return props.color || _color.black;
}, defaultMargin);

exports.H4 = H4;

var H5 = _styledComponents["default"].h5.withConfig({
  displayName: "Heading__H5",
  componentId: "xu06vv-4"
})(["font-size:", ";line-height:", ";font-weight:", ";color:", ";", ""], function (props) {
  return props.fontSize || _typography.fontSizes.z500;
}, _typography.lineHeight.regular, _typography.fontWeights.regular, function (props) {
  return props.color || _color.black;
}, defaultMargin);

exports.H5 = H5;

var H6 = _styledComponents["default"].h6.withConfig({
  displayName: "Heading__H6",
  componentId: "xu06vv-5"
})(["font-size:", ";line-height:", ";font-weight:", ";color:", ";", ""], function (props) {
  return props.fontSize || _typography.fontSizes.z400;
}, _typography.lineHeight.regular, _typography.fontWeights.regular, function (props) {
  return props.color || _color.black;
}, defaultMargin);

exports.H6 = H6;
