// webuikit/esm/Typography/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "H1", {
  enumerable: true,
  get: function get() {
    return _Heading.H1;
  }
});
Object.defineProperty(exports, "H2", {
  enumerable: true,
  get: function get() {
    return _Heading.H2;
  }
});
Object.defineProperty(exports, "H3", {
  enumerable: true,
  get: function get() {
    return _Heading.H3;
  }
});
Object.defineProperty(exports, "H4", {
  enumerable: true,
  get: function get() {
    return _Heading.H4;
  }
});
Object.defineProperty(exports, "H5", {
  enumerable: true,
  get: function get() {
    return _Heading.H5;
  }
});
Object.defineProperty(exports, "H6", {
  enumerable: true,
  get: function get() {
    return _Heading.H6;
  }
});
Object.defineProperty(exports, "Mark", {
  enumerable: true,
  get: function get() {
    return _Mark["default"];
  }
});
Object.defineProperty(exports, "Text", {
  enumerable: true,
  get: function get() {
    return _Text["default"];
  }
});
Object.defineProperty(exports, "P", {
  enumerable: true,
  get: function get() {
    return _Text.P;
  }
});
Object.defineProperty(exports, "Strong", {
  enumerable: true,
  get: function get() {
    return _Text.Strong;
  }
});
Object.defineProperty(exports, "Underline", {
  enumerable: true,
  get: function get() {
    return _Text.Underline;
  }
});
Object.defineProperty(exports, "Tag", {
  enumerable: true,
  get: function get() {
    return _Text.Tag;
  }
});

var _Heading = require("./Heading");

var _Mark = _interopRequireDefault(require("./Mark"));

var _Text = _interopRequireWildcard(require("./Text"));
