// webuikit/esm/Modal/Modal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Portal = _interopRequireDefault(require("../HOC/Portal"));

var _Button = _interopRequireDefault(require("../Button"));

var _Icon = _interopRequireDefault(require("../Icon"));

var _color = require("../tokens/color");

var _lodashEs = require("lodash-es");

var _misc = _interopRequireDefault(require("../tokens/misc"));

var _commonHelper = require("../helpers/commonHelper");

var modalWidth = {
  small: "20vw",
  medium: "40vw",
  large: "60vw",
  "default": "max-content"
};

var Modal = function Modal(_ref) {
  var _ref$onClose = _ref.onClose,
      onClose = _ref$onClose === void 0 ? _lodashEs.noop : _ref$onClose,
      _ref$visible = _ref.visible,
      visible = _ref$visible === void 0 ? false : _ref$visible,
      _ref$type = _ref.type,
      type = _ref$type === void 0 ? "default" : _ref$type,
      _ref$children = _ref.children,
      children = _ref$children === void 0 ? null : _ref$children,
      _ref$isDialog = _ref.isDialog,
      isDialog = _ref$isDialog === void 0 ? false : _ref$isDialog,
      _ref$onAccept = _ref.onAccept,
      onAccept = _ref$onAccept === void 0 ? _lodashEs.noop : _ref$onAccept,
      _ref$onReject = _ref.onReject,
      onReject = _ref$onReject === void 0 ? _lodashEs.noop : _ref$onReject,
      _ref$acceptLabel = _ref.acceptLabel,
      acceptLabel = _ref$acceptLabel === void 0 ? "" : _ref$acceptLabel,
      _ref$rejectLabel = _ref.rejectLabel,
      rejectLabel = _ref$rejectLabel === void 0 ? "" : _ref$rejectLabel,
      _ref$title = _ref.title,
      title = _ref$title === void 0 ? "" : _ref$title,
      _ref$showCloseIcon = _ref.showCloseIcon,
      showCloseIcon = _ref$showCloseIcon === void 0 ? true : _ref$showCloseIcon;

  var handleCloseClick = function handleCloseClick(e) {
    e.preventDefault();
    e.stopPropagation();
    onClose();
  };

  var suppressClick = function suppressClick(e) {
    e.preventDefault();
    e.stopPropagation();
  };

  (0, _react.useEffect)(function () {
    visible ? document.body.style.overflow = "hidden" : document.body.style.overflow = "auto";
  }, [visible]);
  return _react["default"].createElement(_Portal["default"], null, _react["default"].createElement(Modal.Wrapper, {
    "aria-hidden": visible ? "false" : "true",
    role: "dialog",
    visible: visible
  }, _react["default"].createElement(Modal.Overlay, {
    onClick: handleCloseClick,
    visible: visible,
    tabIndex: "-1"
  }), _react["default"].createElement(ModalContainer, {
    onClick: suppressClick,
    type: type,
    visible: visible,
    isDialog: isDialog
  }, (!isDialog && !!showCloseIcon || !!title) && _react["default"].createElement(ModalContainer.Head, null, _react["default"].createElement(ModalContainer.Title, null, title), showCloseIcon && _react["default"].createElement(_Icon["default"], {
    name: "cross",
    size: 24,
    showPointer: true,
    onClick: handleCloseClick,
    onKeyUp: (0, _commonHelper.onKeySelect)(handleCloseClick),
    tabIndex: "0",
    "aria-label": "close Modal"
  })), _react["default"].createElement(ModalContainer.Content, {
    visible: visible
  }, children), isDialog && _react["default"].createElement(ActionButtons, {
    rejectLabel: rejectLabel,
    acceptLabel: acceptLabel,
    onReject: onReject,
    onAccept: onAccept
  }))));
};

var ActionButtons = function ActionButtons(props) {
  var _props$rejectLabel = props.rejectLabel,
      rejectLabel = _props$rejectLabel === void 0 ? "No" : _props$rejectLabel,
      _props$acceptLabel = props.acceptLabel,
      acceptLabel = _props$acceptLabel === void 0 ? "Yes" : _props$acceptLabel,
      _props$onReject = props.onReject,
      onReject = _props$onReject === void 0 ? _lodashEs.noop : _props$onReject,
      _props$onAccept = props.onAccept,
      onAccept = _props$onAccept === void 0 ? _lodashEs.noop : _props$onAccept;
  return _react["default"].createElement(ActionButtons.Wrapper, null, _react["default"].createElement(ActionButtons.Holder, null, _react["default"].createElement(_Button["default"], {
    appearance: "link",
    btnColor: "red",
    size: "small",
    onKeyUp: (0, _commonHelper.onKeySelect)(onReject),
    onClick: onReject,
    tabIndex: "0",
    role: "button"
  }, rejectLabel), _react["default"].createElement(ActionButtons.Gap, null), _react["default"].createElement(_Button["default"], {
    btnColor: "red",
    size: "small",
    onClick: onAccept,
    onKeyUp: (0, _commonHelper.onKeySelect)(onAccept),
    tabIndex: "0",
    role: "button"
  }, acceptLabel)));
};

ActionButtons.propTypes = {
  rejectLabel: _propTypes["default"].string,
  acceptLabel: _propTypes["default"].string,
  onAccept: _propTypes["default"].func,
  onReject: _propTypes["default"].func
};
ActionButtons.Holder = _styledComponents["default"].div.withConfig({
  displayName: "Modal__Holder",
  componentId: "sc-1gr0s0a-0"
})(["display:flex;justify-content:flex-end;width:100%;min-width:max-content;margin-top:1rem;"]);
ActionButtons.Gap = _styledComponents["default"].div.withConfig({
  displayName: "Modal__Gap",
  componentId: "sc-1gr0s0a-1"
})(["height:1rem;width:1rem;"]);
ActionButtons.Wrapper = _styledComponents["default"].div.withConfig({
  displayName: "Modal__Wrapper",
  componentId: "sc-1gr0s0a-2"
})(["display:flex;"]);
Modal.Wrapper = _styledComponents["default"].div.withConfig({
  displayName: "Modal__Wrapper",
  componentId: "sc-1gr0s0a-3"
})(["position:fixed;top:0;left:0;width:100%;min-width:max-content;height:100%;display:flex;justify-content:center;visibility:", ";overflow:auto;padding:5rem 0;"], function (props) {
  return props.visible ? "visible" : "hidden";
});
Modal.propTypes = {
  children: _propTypes["default"].node,
  onClose: _propTypes["default"].func,
  visible: _propTypes["default"].bool,
  type: _propTypes["default"].string,
  isDialog: _propTypes["default"].bool,
  onAccept: _propTypes["default"].func,
  onReject: _propTypes["default"].func,
  rejectLabel: _propTypes["default"].string,
  acceptLabel: _propTypes["default"].string,
  title: _propTypes["default"].string,
  showCloseIcon: _propTypes["default"].bool
};
Modal.defaultProps = {
  children: null
};

var getModalWidth = function getModalWidth(_ref2) {
  var type = _ref2.type;
  return modalWidth[type];
};

Modal.Overlay = _styledComponents["default"].div.withConfig({
  displayName: "Modal__Overlay",
  componentId: "sc-1gr0s0a-4"
})(["position:fixed;top:0;left:0;width:100%;height:100%;background-color:", ";z-index:5;will-change:opacity;opacity:", ";transition:opacity ", " ease;"], _color.black, function (props) {
  return props.visible ? 0.85 : 0;
}, _misc["default"].animationDuration);

var ModalContainer = _styledComponents["default"].div.withConfig({
  displayName: "Modal__ModalContainer",
  componentId: "sc-1gr0s0a-5"
})(["width:", ";height:max-content;margin:auto;min-height:", ";min-width:20rem;background:#fff;border-radius:0.6rem;z-index:10;opacity:1;transform:scale(", ");transition:transform ", " ease;display:", ";box-shadow:0 1.2rem 7rem rgba(28,28,28,0.15);padding:", ";@media (max-width:480px){width:95vw;}"], getModalWidth, function (props) {
  return props.isDialog ? 0 : "150px";
}, function (props) {
  return props.visible ? 1 : 0.8;
}, _misc["default"].animationDuration, function (props) {
  return props.visible ? "block" : "flex";
}, function (props) {
  return props.isDialog ? "2rem" : 0;
});

ModalContainer.Head = _styledComponents["default"].section.withConfig({
  displayName: "Modal__Head",
  componentId: "sc-1gr0s0a-6"
})(["width:calc(100% - 4.6rem);display:flex;justify-content:space-between;margin:2rem 2.2rem 0 2.4rem;@media (max-width:480px){width:calc(100% - 2.6rem);margin:1rem 1.2rem 0 1.4rem;}"]);
ModalContainer.Title = _styledComponents["default"].h2.withConfig({
  displayName: "Modal__Title",
  componentId: "sc-1gr0s0a-7"
})(["font-size:2.4rem;line-height:2.8rem;font-weight:400;color:", ";"], _color.black);
ModalContainer.Content = _styledComponents["default"].section.withConfig({
  displayName: "Modal__Content",
  componentId: "sc-1gr0s0a-8"
})(["display:", ";@media (max-width:480px){min-height:8rem;}"], function (props) {
  return props.visible ? "block" : "none";
});
Modal.propTypes = {
  type: _propTypes["default"].oneOf(["small", "medium", "large", "default"]),
  children: _propTypes["default"].node,
  onClose: _propTypes["default"].func.isRequired
};
var _default = Modal;
exports["default"] = _default;
