// webuikit/es/MenuItem/MenuItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Icon = _interopRequireDefault(require("../Icon"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _color = require("../tokens/color");

var _lodashEs = require("lodash-es");

var MenuItem = function MenuItem(props) {
  var _props$leftIcon = props.leftIcon,
      leftIcon = _props$leftIcon === void 0 ? "" : _props$leftIcon,
      _props$value = props.value,
      value = _props$value === void 0 ? "" : _props$value,
      _props$label = props.label,
      label = _props$label === void 0 ? "" : _props$label,
      _props$desc = props.desc,
      desc = _props$desc === void 0 ? "" : _props$desc,
      _props$selected = props.selected,
      selected = _props$selected === void 0 ? false : _props$selected,
      _props$bold = props.bold,
      bold = _props$bold === void 0 ? "" : _props$bold,
      _props$id = props.id,
      id = _props$id === void 0 ? "" : _props$id,
      _passClick = props._passClick;

  var handleClick = function handleClick(value, label) {
    return function (e) {
      e.preventDefault();

      _passClick(value, label);
    };
  };

  return _react["default"].createElement(MenuItem.Wrapper, {
    onClick: handleClick(value, label),
    isSelected: selected,
    id: selected ? id + "-selected" : id + "-" + value
  }, _react["default"].createElement(LeftSection, null, _react["default"].createElement(_Icon["default"], {
    name: leftIcon,
    size: 16,
    color: _color.grey.z800
  }), _react["default"].createElement(Span, {
    isSelected: selected
  }, getBoldLabel(label, bold))), !!desc && _react["default"].createElement(DescSpan, null, desc));
};

var getBoldLabel = function getBoldLabel(label, str) {
  if (!str) return label;
  var pattern = new RegExp(str, "i");
  var match = label.match(pattern);
  label = label.replace(pattern, "");
  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement(Bold, null, match), label);
};

var Bold = _styledComponents["default"].span.withConfig({
  displayName: "MenuItem__Bold",
  componentId: "sc-15nkcye-0"
})(["font-weight:500;font-size:inherit;color:", ";"], _color.black);

MenuItem.propTypes = {
  leftIcon: _propTypes["default"].string,
  text: _propTypes["default"].string,
  label: _propTypes["default"].string,
  desc: _propTypes["default"].string,
  value: _propTypes["default"].string,
  selected: _propTypes["default"].bool,
  bold: _propTypes["default"].string,
  id: _propTypes["default"].string,
  _passClick: _lodashEs.noop
};

var DescSpan = _styledComponents["default"].span.withConfig({
  displayName: "MenuItem__DescSpan",
  componentId: "sc-15nkcye-1"
})(["font-size:1.5rem;color:", ";white-space:nowrap;margin-left:3rem;overflow:hidden;text-overflow:ellipsis;font-weight:300;"], _color.grey.z400);

var LeftSection = _styledComponents["default"].span.withConfig({
  displayName: "MenuItem__LeftSection",
  componentId: "sc-15nkcye-2"
})(["display:flex;"]);

var Span = _styledComponents["default"].span.withConfig({
  displayName: "MenuItem__Span",
  componentId: "sc-15nkcye-3"
})(["margin-left:1rem;margin-top:0.3rem;font-size:1.8rem;color:", ";font-weight:300;"], _color.grey.z800);

MenuItem.Wrapper = _styledComponents["default"].div.withConfig({
  displayName: "MenuItem__Wrapper",
  componentId: "sc-15nkcye-4"
})(["display:flex;align-items:center;justify-content:space-between;width:100%;padding:1.3rem 2rem;cursor:pointer;background-color:", ";&:hover{background-color:", ";}&:hover > span{color:", ";}"], function (props) {
  return props.isSelected ? "#E9F2FC" : "initial";
}, _color.grey.z100, _color.grey.z800);
var _default = MenuItem;
exports["default"] = _default;
