// webuikit/es/Typography/Text.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Tag = exports.Underline = exports.Strong = exports.P = exports["default"] = void 0;

var _typography = require("../tokens/typography");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _color = require("../tokens/color");

//p, span, bold,strong,pre,code,small,Label
// eslint-disable-next-line no-unused-vars
var Text = function Text(_) {
  return "";
};

Text.p = _styledComponents["default"].p.withConfig({
  displayName: "Text__p",
  componentId: "sc-1qii6o5-0"
})(["font-size:", ";line-height:", ";margin:", " 0;color:", ";& > a{display:inline-block;color:", ";text-decoration:none;&:hover{color:", ";}}"], function (props) {
  return props.size || "inherit";
}, _typography.lineHeight.medium, _typography.fontSizes.z300, _color.grey.z900, _color.red.z400, _color.red.z500);
Text.strong = _styledComponents["default"].strong.withConfig({
  displayName: "Text__strong",
  componentId: "sc-1qii6o5-1"
})(["line-height:", ";font-weight:", ";color:", ";font-size:inherit;"], _typography.lineHeight.medium, _typography.fontWeights.medium, _color.grey.z800);
Text.underline = _styledComponents["default"].span.withConfig({
  displayName: "Text__underline",
  componentId: "sc-1qii6o5-2"
})(["color:", ";font-size:inherit;text-decoration:underline;"], _color.grey.z800);
Text.Tag = _styledComponents["default"].a.withConfig({
  displayName: "Text__Tag",
  componentId: "sc-1qii6o5-3"
})(["display:inline;font-size:24px;line-height:28px;color:", ";cursor:pointer;margin-top:2rem;margin-bottom:1rem;text-transform:capitalize;&:hover{color:", ";}"], _color.red.z400, _color.red.z500);
Text.LargeP = _styledComponents["default"].p.withConfig({
  displayName: "Text__LargeP",
  componentId: "sc-1qii6o5-4"
})(["font-size:24px;line-height:28px;margin:", " 0;color:", ";"], _typography.fontSizes.z300, _color.grey.z900);
var P = Text.p;
exports.P = P;
var Strong = Text.strong;
exports.Strong = Strong;
var Underline = Text.underline;
exports.Underline = Underline;
var Tag = Text.Tag;
exports.Tag = Tag;
var _default = Text;
exports["default"] = _default;
