// webuikit/tokens/colors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.tag = exports.status = exports.radioLabel = exports.alert = exports.list = exports.icon = exports.imgUploaderDefault = exports.checkboxSelected = exports.checkbox = exports.button = exports.tabs = exports.tooltip = exports.link = exports.input = exports.form = exports.text = exports.base = void 0;

var _color = require("./color");

var base = {
  white: _color.white,
  black: _color.black,
  "default": "#333",
  gray: "#B5B5B5",
  grayDarkest: "#676767",
  grayDark: "#636363",
  grayMedium: "#A8A8A8",
  grayLight: "#D9D9D9",
  grayLightest: "#F1F1F1",
  blueDarker: "#097093",
  blue: "#0a84ae",
  blueLight: "#44c7f4",
  blueLightest: "#f5f7f9",
  orange: "#eb5424",
  orangeLighter: "#ff784d",
  orangeLightest: "#ffb299",
  orangeDark: "#FF3E00",
  greenDarker: "#3F6910",
  green: _color.green.z500,
  greenDark: _color.green.z600,
  greenLight: "",
  red: "#E23744",
  redDarker: "#801F00",
  yellow: "#786600"
};
exports.base = base;
var text = {
  "default": "#212121",
  inputs: "#555",
  secondary: "#757575",
  subtle: "#BDBDBD",
  error: "#FF0000",
  green: "#0DA314"
};
exports.text = text;
var form = {
  title: "#000",
  divider: "rgba(151,151,151,0.29)",
  background: "#f9f9f9"
};
exports.form = form;
var input = {
  "default": {
    background: _color.white,
    borderColor: _color.grey.z300,
    icon: _color.grey.z400,
    text: _color.black,
    placeholder: _color.grey.z400,
    borderWidth: "0.1rem solid",
    helperColor: _color.grey.z600
  },
  focused: {
    background: _color.white,
    borderColor: _color.green.z500,
    icon: _color.grey.z400,
    text: _color.black,
    placeholder: _color.green.z500,
    borderWidth: "0.15rem solid",
    helperColor: _color.grey.z600
  },
  error: {
    background: _color.white,
    borderColor: _color.red.z600,
    icon: _color.grey.z400,
    text: _color.black,
    placeholder: _color.red.z600,
    borderWidth: "0.15rem solid",
    helperColor: _color.red.z600
  },
  disabled: {
    background: _color.white,
    borderColor: _color.grey.z300,
    icon: _color.grey.z400,
    text: _color.grey.z400,
    placeholder: _color.grey.z400,
    borderWidth: "0.1rem solid",
    helperColor: _color.grey.z600
  }
};
exports.input = input;
var link = {
  "default": _color.green.z500,
  defaultHover: _color.green.z600,
  defaultActive: _color.green.z600,
  defaultFocus: _color.green.z600,
  sidebar: _color.green.z600,
  sidebarHover: _color.green.z600,
  sidebarFocus: _color.green.z600
};
exports.link = link;
var tooltip = {
  background: "#1A1A1A",
  text: "#FFF"
};
exports.tooltip = tooltip;
var tabs = {
  "default": "#212121",
  defaultHover: "#0a84ae"
};
exports.tabs = tabs;
var button = {
  "default": {
    background: _color.green.z500,
    backgroundHover: _color.green.z600,
    backgroundFocus: _color.green.z600,
    backgroundActive: _color.green.z600,
    text: _color.white,
    icon: "default",
    border: _color.green.z500,
    borderHover: _color.green.z600,
    borderFocus: _color.green.z600,
    borderActive: _color.green.z600
  },
  outline: {
    background: "#FFF",
    backgroundHover: "#FAFAFA",
    backgroundFocus: "#FAFAFA",
    backgroundActive: "#FAFAFA",
    text: _color.green.z500,
    hoverText: "#3F6910",
    icon: "white",
    border: _color.green.z500,
    borderHover: "#3F6910",
    borderFocus: "#3F6910",
    borderActive: "#3F6910"
  },
  disabled: {
    background: _color.grey.z500,
    backgroundHover: _color.grey.z500,
    backgroundFocus: _color.grey.z500,
    backgroundActive: _color.grey.z500,
    text: _color.white,
    icon: "white",
    border: _color.grey.z500,
    borderHover: _color.grey.z500,
    borderFocus: _color.grey.z500,
    borderActive: _color.grey.z500
  },
  red: {
    background: _color.red.z500,
    backgroundHover: _color.red.z600,
    backgroundFocus: _color.red.z600,
    backgroundActive: _color.red.z600,
    text: _color.white,
    icon: _color.white,
    border: _color.red.z500,
    borderHover: _color.red.z600,
    borderFocus: _color.red.z600,
    borderActive: _color.red.z600
  },
  success: {
    background: "#4CD964",
    backgroundHover: "#4CD964",
    backgroundFocus: "#4CD964",
    text: "#FAFAFA",
    icon: "white",
    border: "#4CD964",
    borderHover: "#4CD964",
    borderFocus: "#4CD964",
    borderActive: "#4CD964"
  },
  link: {
    text: "#0a84ae",
    icon: "blue",
    hover: "#0a84ae",
    focus: "#0a84ae"
  }
};
exports.button = button;
var icon = {
  "default": "#333",
  light: "#A8A8A8",
  sidebar: "default",
  sidebarHover: "orangeDark",
  sidebarFocus: "orangeDark"
};
exports.icon = icon;
var list = {
  backgroundHover: "#FAFAFA",
  borderColor: "#DDD"
};
exports.list = list;
var alert = {
  "default": {
    background: "#F0F0F0",
    text: "#333"
  },
  information: {
    background: "#DBF4FC",
    text: "#097093"
  },
  success: {
    background: "#E5F8D1",
    text: "#3F6910"
  },
  warning: {
    background: "#FFF6CB",
    text: "#786600"
  },
  danger: {
    background: "#FFD8CC",
    text: "#801F00"
  }
};
exports.alert = alert;
var radioLabel = {
  "default": {
    borderColor: _color.grey.z500,
    hoverBorderColor: _color.green.z500,
    selectedBgColor: _color.green.z500
  },
  checked: {
    borderColor: _color.green.z500,
    hoverBorderColor: _color.green.z500,
    selectedBgColor: _color.green.z500
  },
  disabled: {
    borderColor: _color.grey.z300,
    hoverBorderColor: _color.grey.z300,
    selectedBgColor: _color.grey.z300
  }
};
exports.radioLabel = radioLabel;
var checkboxSelected = {
  disabled_solid: {
    borderColor: _color.grey.z300,
    background: _color.grey.z300,
    tickColor: _color.white
  },
  disabled: {
    borderColor: _color.grey.z300,
    tickColor: _color.grey.z300,
    background: _color.white
  },
  solid: {
    borderColor: _color.green.z500,
    background: _color.green.z500,
    tickColor: _color.white
  },
  "default": {
    borderColor: _color.green.z500,
    tickColor: _color.green.z500,
    background: _color.white
  }
};
exports.checkboxSelected = checkboxSelected;
var checkbox = {
  disabled: {
    borderColor: "#CFCFCF",
    hoverBorderColor: "#CFCFCF"
  },
  "default": {
    borderColor: "#9C9C9C",
    hoverBorderColor: "#28A745"
  }
};
exports.checkbox = checkbox;
var imgUploaderDefault = {
  tagColor: _color.grey.z500,
  borderColor: _color.grey.z300,
  background: _color.white
};
exports.imgUploaderDefault = imgUploaderDefault;
var status = {
  "default": "#7D7D7D",
  information: "#3BC0F2",
  success: "#73CD1F",
  warning: "#F49C20",
  danger: "#D14B49"
};
exports.status = status;
var tag = {
  background: "#DBF4FD",
  border: "#3BC0F2",
  text: "#3BC0F2"
};
exports.tag = tag;
