// webuikit/es/Flag/Flag.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _commonHelper = require("../helpers/commonHelper");

var DEFAULT_WIDTH = 24;
var DEFAULT_XY = "0 0";
var FLAGS_SPRITE_URL = "data/webuikit/39194387e4bcd5ee03cc42eb19e6febf1556092738.png";

var Flag = function Flag(props) {
  var _props$width = props.width,
      width = _props$width === void 0 ? DEFAULT_WIDTH : _props$width,
      _props$country = props.country,
      country = _props$country === void 0 ? "" : _props$country;
  var xy = getXY(country);
  var scale = getScaleFactor(DEFAULT_WIDTH, width);
  var leftMargin = getLeftMargin(scale, DEFAULT_WIDTH);
  return _react["default"].createElement(Flag.Image, {
    width: DEFAULT_WIDTH,
    xy: xy,
    scaleFactor: scale,
    leftMargin: leftMargin
  });
};

var getLeftMargin = function getLeftMargin(scale, defaultWidth) {
  scale === 1 ? 0 : defaultWidth * scale;
  if (!scale) return 0; // TODO refactor this code

  return Math.floor((defaultWidth * scale - defaultWidth) / 2);
};

var getScaleFactor = function getScaleFactor(defaultWidth, width) {
  return width / defaultWidth;
};

var XY_MAPPING = {
  australia: "-2.4rem 0",
  bahamas: "-9.6rem 0",
  belgium: "-4.8rem 0",
  belize: "-12rem 0",
  brazil: "-7.2rem 0",
  canada: "0 -2.4rem",
  chile: "-2.4rem -2.4rem",
  colombia: "-4.8rem -2.4rem",
  "czech-republic": "-7.2rem -2.4rem",
  denmark: "-9.6rem -2.4rem",
  finland: "-12rem -2.4rem",
  india: "-4.8rem -4.8rem",
  indonesia: "0 -4.8rem",
  ireland: "-2.4rem -4.8rem",
  italy: "-7.2rem -4.8rem",
  lebanon: "-9.6rem -4.8rem",
  luxembourg: "-12rem -4.8rem",
  malaysia: "-2.4rem -7.2rem",
  mauritius: "0 -7.2rem",
  netherlands: "-4.8rem -7.2rem",
  "new-zealand": "-9.6rem -7.2rem",
  norway: "-7.2rem -7.2rem",
  philippines: "-12rem -7.2rem",
  poland: "0 -9.6rem",
  portugal: "-2.4rem -9.6rem"
};

var getXY = function getXY(name) {
  return XY_MAPPING[name] || DEFAULT_XY;
};

Flag.propTypes = {
  width: _propTypes["default"].number,
  country: _propTypes["default"].string
};
Flag.Image = _styledComponents["default"].span.withConfig({
  displayName: "Flag__Image",
  componentId: "sc-1x68twu-0"
})(["width:", "rem;height:", "rem;display:inline-block;background:url(", ") no-repeat;background-position:", ";transform:scale(", ");", ""], function (props) {
  return props.width / 10;
}, function (props) {
  return props.width / 10;
}, (0, _commonHelper.getCDN)(FLAGS_SPRITE_URL), function (props) {
  return props.xy;
}, function (props) {
  return props.scaleFactor;
}, function (props) {
  return !!props.leftMargin && (0, _styledComponents.css)(["margin-left:", "rem;"], props.leftMargin / 10);
});
var _default = Flag;
exports["default"] = _default;
