// webuikit/Link/Link.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _colors = require("../tokens/colors");

var Link = function Link(props) {
  return _react["default"].createElement(Link.Ele, {
    href: props.linkTo,
    target: props.target,
    title: props.title,
    onClick: props.onClick,
    alt: props.alt,
    role: props.role
  }, props.children);
};

Link.Ele = _styledComponents["default"].a.withConfig({
  displayName: "Link__Ele",
  componentId: "sc-1glovd4-0"
})(["font-weight:300;color:", ";text-decoration:none;cursor:pointer;&:hover{color:", ";}&:active{color:", ";}"], _colors.link["default"], _colors.link.defaultHover, _colors.link.defaultActive);
Link.propTypes = {
  linkTo: _propTypes["default"].string,
  target: _propTypes["default"].oneOf(["_self", "_blank", "_parent", "_top"]),
  title: _propTypes["default"].string,
  onClick: _propTypes["default"].func,

  /** for accessebility */
  alt: _propTypes["default"].string.isRequired,
  role: _propTypes["default"].string
};
Link.defaultProps = {
  target: "_self",
  role: "link"
};
var _default = Link;
exports["default"] = _default;
