// webuikit/esm/HOC/OnClickOutside.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var OnClickOutside = function OnClickOutside(props) {
  var node = (0, _react.useRef)();
  (0, _react.useEffect)(function () {
    document.addEventListener("mousedown", handleOutsideClick);
    return function () {
      document.removeEventListener("mousedown", handleOutsideClick);
    };
  });

  var handleOutsideClick = function handleOutsideClick(e) {
    if (node.current && node.current.contains(e.target)) {
      return;
    }

    props.onClickOutside();
  };

  return _react["default"].createElement(OnClickOutside.Wrapper, {
    ref: node
  }, props.children);
};

OnClickOutside.Wrapper = _styledComponents["default"].div.withConfig({
  displayName: "OnClickOutside__Wrapper",
  componentId: "sc-12qhdnm-0"
})(["width:max-content;"]);
OnClickOutside.propTypes = {
  children: _propTypes["default"].node,
  onClickOutside: _propTypes["default"].func.isRequired
};
var _default = OnClickOutside; // USAGE:

/*
 *   <OnClickOutside onClickOutside={handleClickOutside}>
 *     // your component here
 *   </OnClickOutside>
 *
 *   const handleClickOutside = {
 *     // handle outside click
 *   };
 */

exports["default"] = _default;
