// webuikit/RestaurantOrderCard/RestaurantOrderCard.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _ZImage = _interopRequireDefault(require("../ZImage"));

var _constants = require("./constants");

var _orderCardHelpers = require("./orderCardHelpers");

// Libraries
// Components
// Constants
// Helpers
var RestaurantOrderCard = function RestaurantOrderCard(props) {
  return _react["default"].createElement(RestaurantOrderCardWrapper, null, _react["default"].createElement(Img, {
    disabled: props.disabled,
    src: props.imageUrl
  }), _react["default"].createElement(Details, null, _react["default"].createElement(Info, null, _react["default"].createElement("article", null, _react["default"].createElement(Heading, {
    disabled: props.disabled
  }, props.resName), _react["default"].createElement(Type, {
    disabled: props.disabled
  }, props.resType)), _react["default"].createElement("article", null, _react["default"].createElement(Type, {
    disabled: props.disabled
  }, props.timing), _react["default"].createElement(Separator, null), _react["default"].createElement(Type, {
    disabled: props.disabled
  }, props.pricing), (0, _orderCardHelpers.getO2Info)(props))), _react["default"].createElement(Rating, {
    disabled: props.disabled
  }, props.rating)));
};

var RestaurantOrderCardWrapper = _styledComponents["default"].section.withConfig({
  displayName: "RestaurantOrderCard__RestaurantOrderCardWrapper",
  componentId: "sc-3aktam-0"
})(["display:flex;"]);

var Img = (0, _styledComponents["default"])(_ZImage["default"]).withConfig({
  displayName: "RestaurantOrderCard__Img",
  componentId: "sc-3aktam-1"
})(["height:17rem;width:20rem;border-top-left-radius:0.6rem;border-bottom-left-radius:0.6rem;filter:", ";"], (0, _orderCardHelpers.getState)(_constants.IMG_FILTER));

var Heading = _styledComponents["default"].article.withConfig({
  displayName: "RestaurantOrderCard__Heading",
  componentId: "sc-3aktam-2"
})(["font-weight:400;font-size:2rem;line-height:2.9rem;color:", ";"], (0, _orderCardHelpers.getState)(_constants.HEADING_COLOR));

var Details = _styledComponents["default"].section.withConfig({
  displayName: "RestaurantOrderCard__Details",
  componentId: "sc-3aktam-3"
})(["display:flex;width:43rem;padding:1.1rem 1.6rem 1.2rem;border-top:1px solid #e8e8e8;border-right:1px solid #e8e8e8;border-bottom:1px solid #e8e8e8;border-top-right-radius:0.6rem;border-bottom-right-radius:0.6rem;"]);

var Info = _styledComponents["default"].section.withConfig({
  displayName: "RestaurantOrderCard__Info",
  componentId: "sc-3aktam-4"
})(["display:flex;flex-direction:column;justify-content:space-between;height:100%;flex-grow:1;padding-right:2rem;"]);

var Rating = _styledComponents["default"].article.withConfig({
  displayName: "RestaurantOrderCard__Rating",
  componentId: "sc-3aktam-5"
})(["background-color:", ";border-radius:0.6rem;font-size:2rem;line-height:2.9rem;padding:0.1rem 0.9rem;height:fit-content;color:#fff;"], (0, _orderCardHelpers.getState)(_constants.RATING_COLOR));

var Type = _styledComponents["default"].span.withConfig({
  displayName: "RestaurantOrderCard__Type",
  componentId: "sc-3aktam-6"
})(["color:", ";padding-top:0.3rem;font-size:1.6rem;line-height:2.5rem;"], (0, _orderCardHelpers.getState)(_constants.TYPE_COLOR));

var Separator = _styledComponents["default"].span.withConfig({
  displayName: "RestaurantOrderCard__Separator",
  componentId: "sc-3aktam-7"
})(["width:1px;background-color:#cfcfcf;height:100%;border:0.5px solid #cfcfcf;margin:0 0.9rem;border-radius:2px;"]);

RestaurantOrderCard.propTypes = {
  imageUrl: _propTypes["default"].string,
  resName: _propTypes["default"].string.isRequired,
  resType: _propTypes["default"].string.isRequired,
  timing: _propTypes["default"].string.isRequired,
  pricing: _propTypes["default"].string.isRequired,
  rating: _propTypes["default"].string.isRequired,
  discount: _propTypes["default"].string,
  codeStatement: _propTypes["default"].string,
  disabledStatement: _propTypes["default"].string,
  disabled: _propTypes["default"].bool
};
RestaurantOrderCard.defaultProps = {
  discount: "",
  codeStatement: "",
  disabledStatement: "",
  disabled: false
};
var _default = RestaurantOrderCard;
exports["default"] = _default;
