// webuikit/Slider/Slider.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _reactInputRange = _interopRequireDefault(require("react-input-range"));

var _lodashEs = require("lodash-es");

require("style-loader!react-input-range/lib/css/index.css");

// Libraries
// styles
var RangeWrapper = _styledComponents["default"].section.withConfig({
  displayName: "Slider__RangeWrapper",
  componentId: "sc-16zxror-0"
})([".input-range__track--active{background:#f25a65;}.input-range__slider{height:2.2rem;width:2.2rem;top:-0.65rem;background:#f25a65;border:2px solid #ffffff;}.input-range__slider:hover,.input-range__slider:active{}.input-range__label--value{top:-2.5rem;}.input-range__label--min,{.input-range__label--max{display:none !important;}.input-range__label--value .input-range__label-container{bottom:2.9rem;background:#000;opacity:0.8;padding:7px;left:-0.8rem;border-radius:3px;color:white;display:flex;&:after{content:\"\";position:absolute;top:100%;left:50%;margin-left:-5px;border-width:5px;border-style:solid;border-color:#000 transparent transparent transparent;}}"]);

var Slider = function Slider(props) {
  var _useState = (0, _react.useState)(props.value),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      value = _useState2[0],
      setValue = _useState2[1];

  var onChange = function onChange(newValue) {
    props.onChange(newValue);
    setValue(newValue);
  };

  return _react["default"].createElement(RangeWrapper, null, _react["default"].createElement(_reactInputRange["default"], {
    maxValue: props.maxValue,
    minValue: props.minValue,
    value: value,
    onChange: onChange,
    formatLabel: props.formatLabel
  }));
};

Slider.propTypes = {
  minValue: _propTypes["default"].number,
  maxValue: _propTypes["default"].number,
  value: _propTypes["default"].object.isRequired,
  onChange: _propTypes["default"].func,
  formatLabel: _propTypes["default"].func
};
Slider.defaultProps = {
  minValue: 0,
  maxValue: 10,
  onChange: _lodashEs.noop,
  formatLabel: function formatLabel(value) {
    return "".concat(value);
  }
};
var _default = Slider;
exports["default"] = _default;
