// webuikit/RadioGroup/RadioGroup.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _lodashEs = require("lodash-es");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Radio = _interopRequireDefault(require("../Radio"));

var _radiogroupHelpers = require("./radiogroupHelpers");

// libraries
// components
// helpers
var RadioGroup = function RadioGroup(props) {
  var _useState = (0, _react.useState)(props.defaultValue),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      selected = _useState2[0],
      setValue = _useState2[1];

  (0, _react.useEffect)(function () {
    props.onChange(selected);
  });

  var onChange = function onChange(e) {
    setValue(e.target.value);
  };

  return _react["default"].createElement(RadioGroup.Element, {
    onChange: onChange,
    horizontal: props.horizontal
  }, props.options.map(function (_ref) {
    var label = _ref.label,
        value = _ref.value,
        disabled = _ref.disabled;
    return _react["default"].createElement(_Radio["default"], {
      key: props.name,
      name: props.name,
      value: value,
      label: label,
      checked: value === selected,
      disabled: !!disabled
    });
  }));
};

RadioGroup.Element = _styledComponents["default"].section.withConfig({
  displayName: "RadioGroup__Element",
  componentId: "sc-1hm8vf5-0"
})(["display:", ";justify-content:", ";"], (0, _radiogroupHelpers.getAttributes)("display"), (0, _radiogroupHelpers.getAttributes)("justifyContent"));
RadioGroup.propTypes = {
  onChange: _propTypes["default"].func,
  horizontal: _propTypes["default"].bool,
  defaultValue: _propTypes["default"].string,
  name: _propTypes["default"].string.isRequired,
  options: _propTypes["default"].array.isRequired
};
RadioGroup.defaultProps = {
  onChange: _lodashEs.noop,
  horizontal: false
};
var _default = RadioGroup;
exports["default"] = _default;
