// webuikit/esm/Searchbox/Textfield.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _color = require("../tokens/color");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _lodashEs = require("lodash-es");

var _Icon = _interopRequireDefault(require("../Icon"));

var _misc = _interopRequireDefault(require("../tokens/misc"));

var Textfield = function Textfield(props) {
  var _props$placeholder = props.placeholder,
      placeholder = _props$placeholder === void 0 ? "" : _props$placeholder,
      _props$searchStr = props.searchStr,
      searchStr = _props$searchStr === void 0 ? "" : _props$searchStr,
      _props$onChange = props.onChange,
      onChange = _props$onChange === void 0 ? _lodashEs.noop : _props$onChange,
      _props$onRemove = props.onRemove,
      onRemove = _props$onRemove === void 0 ? _lodashEs.noop : _props$onRemove,
      _props$value = props.value,
      value = _props$value === void 0 ? "" : _props$value,
      _props$tags = props.tags,
      tags = _props$tags === void 0 ? [] : _props$tags,
      _props$type = props.type,
      type = _props$type === void 0 ? "default" : _props$type,
      _props$setInputRef = props.setInputRef,
      setInputRef = _props$setInputRef === void 0 ? _lodashEs.noop : _props$setInputRef;
  var node = (0, _react.useRef)();
  (0, _react.useEffect)(function () {
    if (tags.length) placeCursor();
  });
  (0, _react.useEffect)(function () {
    if (node.current) setInputRef(node);
  }, [setInputRef]);

  var placeCursor = function placeCursor() {
    return node.current.focus();
  };

  var getWidth = function getWidth() {
    return (searchStr.length + 1) * 12 + "px";
  };

  var removeItem = function removeItem(item) {
    return function () {
      return onRemove(item);
    };
  };

  var showTags = function showTags(tags) {
    return tags.map(function (t) {
      return _react["default"].createElement(TagSpan, {
        key: t,
        background: getBackground(type),
        color: getColor(type)
      }, _react["default"].createElement(Span, null, t), _react["default"].createElement(_Icon["default"], {
        name: "cross",
        showPointer: true,
        color: getColor(type),
        onClick: removeItem(t)
      }));
    });
  };

  return _react["default"].createElement(Textfield.Wrapper, {
    onClick: placeCursor,
    hasTag: !!tags.length
  }, !searchStr && !tags.length && _react["default"].createElement(Textfield.Placeholder, {
    onClick: placeCursor
  }, placeholder), showTags(tags), _react["default"].createElement(Textfield.Input, {
    tabIndex: "0",
    ref: node,
    value: value,
    onChange: onChange,
    hasTag: !!tags.length,
    hasSearchStr: !!searchStr,
    setWidth: getWidth
  }));
};

var getColor = function getColor(type) {
  return colors[type || "default"] || _color.black;
};

var getBackground = function getBackground(type) {
  return backgrounds[type || "default"] || _color.grey.z200;
};

var colors = {
  "default": _color.black,
  red: _color.white,
  green: _color.white
};
var backgrounds = {
  "default": _color.grey.z200,
  red: _color.red.z500,
  green: _color.green.z500
};

var Span = _styledComponents["default"].span.withConfig({
  displayName: "Textfield__Span",
  componentId: "sc-1jqajcj-0"
})(["margin-right:1rem;margin-top:0.2rem;font-size:1.8rem;font-weight:300;"]);

var TagSpan = _styledComponents["default"].span.withConfig({
  displayName: "Textfield__TagSpan",
  componentId: "sc-1jqajcj-1"
})(["position:relative;display:inline-flex;float:left;padding:1.05rem 1rem 1.1rem 1.5rem;background:", ";color:", ";border-radius:", ";margin:0.6rem 0.6rem 0 0;"], function (props) {
  return props.background;
}, function (props) {
  return props.color;
}, _misc["default"].radius);

Textfield.propTypes = {
  searchStr: _propTypes["default"].string,
  placeholder: _propTypes["default"].string,
  onChange: _propTypes["default"].func.isRequired,
  onRemove: _propTypes["default"].func.isRequired,
  value: _propTypes["default"].string,
  tags: _propTypes["default"].array,
  type: _propTypes["default"].oneOf(["default", "red", "green"]),
  setInputRef: _propTypes["default"].func.isRequired
};
Textfield.Wrapper = _styledComponents["default"].div.withConfig({
  displayName: "Textfield__Wrapper",
  componentId: "sc-1jqajcj-2"
})(["position:relative;margin-bottom:", ";overflow:auto;padding:", ";"], function (props) {
  return props.hasTag ? "0.5rem" : "0";
}, function (props) {
  return props.hasTag ? "0" : "0.1rem 0";
});
Textfield.Placeholder = _styledComponents["default"].div.withConfig({
  displayName: "Textfield__Placeholder",
  componentId: "sc-1jqajcj-3"
})(["position:absolute;cursor:text;color:", ";font-weight:300;font-size:1.8rem;margin-top:0.2rem;"], _color.grey.z400);
Textfield.Input = _styledComponents["default"].input.withConfig({
  displayName: "Textfield__Input",
  componentId: "sc-1jqajcj-4"
})(["color:", ";width:", ";overflow:hidden;text-overflow:ellipsis;outline:none;height:", ";border:none;font-family:inherit;margin-top:", ";font-size:18px;", ""], _color.black, function (props) {
  return props.hasSearchStr ? props.setWidth : "1rem";
}, function (props) {
  return props.hasTag ? "4.2rem" : "100%";
}, function (props) {
  return props.hasTag ? "0.6rem" : "0";
}, function (props) {
  return props.hasTag && (0, _styledComponents.css)(["float:left;"]);
});
var _default = Textfield;
exports["default"] = _default;
