// webuikit/esm/Dialog/Dialog.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _Modal = _interopRequireDefault(require("../Modal"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _lodashEs = require("lodash-es");

var Dialog = function Dialog(props) {
  var _props$onAccept = props.onAccept,
      onAccept = _props$onAccept === void 0 ? _lodashEs.noop : _props$onAccept,
      _props$onReject = props.onReject,
      onReject = _props$onReject === void 0 ? _lodashEs.noop : _props$onReject,
      _props$rejectLabel = props.rejectLabel,
      rejectLabel = _props$rejectLabel === void 0 ? "" : _props$rejectLabel,
      _props$acceptLabel = props.acceptLabel,
      acceptLabel = _props$acceptLabel === void 0 ? "" : _props$acceptLabel;
  return _react["default"].createElement(_Modal["default"], (0, _extends2["default"])({}, props, {
    isDialog: true,
    type: "medium",
    onAccept: onAccept,
    onReject: onReject,
    rejectLabel: rejectLabel,
    acceptLabel: acceptLabel
  }));
};

Dialog.propTypes = {
  onAccept: _propTypes["default"].func,
  onReject: _propTypes["default"].func,
  rejectLabel: _propTypes["default"].string,
  acceptLabel: _propTypes["default"].string
};
var _default = Dialog;
exports["default"] = _default;
