// webuikit/es/AvatarCard/AvatarCard.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _lodashEs = require("lodash-es");

var _Icon = _interopRequireDefault(require("../Icon"));

var _color = require("../tokens/color");

// Libraries
// Components
// Helpers
var AvatarCard = function AvatarCard(props) {
  var _props$imgSrc = props.imgSrc,
      imgSrc = _props$imgSrc === void 0 ? "" : _props$imgSrc,
      _props$imageWidth = props.imageWidth,
      imageWidth = _props$imageWidth === void 0 ? "100%" : _props$imageWidth,
      _props$name = props.name,
      name = _props$name === void 0 ? "" : _props$name,
      _props$designation = props.designation,
      designation = _props$designation === void 0 ? "" : _props$designation,
      _props$linkText = props.linkText,
      linkText = _props$linkText === void 0 ? "" : _props$linkText,
      _props$link = props.link,
      link = _props$link === void 0 ? "" : _props$link,
      _props$className = props.className,
      className = _props$className === void 0 ? "" : _props$className;
  return _react["default"].createElement(AvatarCard.Wrapper, {
    className: className
  }, _react["default"].createElement("img", {
    src: imgSrc,
    width: imageWidth
  }), _react["default"].createElement(AvatarCard.Name, null, name), _react["default"].createElement(AvatarCard.Designation, null, designation), _react["default"].createElement(AvatarCard.Link, {
    href: link
  }, _react["default"].createElement(Span, null, linkText), _react["default"].createElement(LinkIcon, {
    name: "back",
    color: _color.red.z400,
    size: 10
  })));
};

AvatarCard.propTypes = {
  imgSrc: _propTypes["default"].string,
  imageWidth: _propTypes["default"].string,
  name: _propTypes["default"].string,
  designation: _propTypes["default"].string,
  linkText: _propTypes["default"].string,
  link: _propTypes["default"].string,
  className: _propTypes["default"].string
};
var LinkIcon = (0, _styledComponents["default"])(_Icon["default"]).withConfig({
  displayName: "AvatarCard__LinkIcon",
  componentId: "sc-1knyk5x-0"
})(["transform:rotate(180deg);"]);
AvatarCard.Name = _styledComponents["default"].p.withConfig({
  displayName: "AvatarCard__Name",
  componentId: "sc-1knyk5x-1"
})(["font-size:2.4rem;color:", ";line-height:2.8rem;margin:2rem 0 0 0;weight:400;"], _color.black);
AvatarCard.Designation = _styledComponents["default"].p.withConfig({
  displayName: "AvatarCard__Designation",
  componentId: "sc-1knyk5x-2"
})(["font-size:1.5rem;line-height:1.8rem;margin:0.5rem 0;color:", ";"], _color.black);
AvatarCard.Link = _styledComponents["default"].a.withConfig({
  displayName: "AvatarCard__Link",
  componentId: "sc-1knyk5x-3"
})(["display:inline-flex;text-decoration:none;"]);

var Span = _styledComponents["default"].span.withConfig({
  displayName: "AvatarCard__Span",
  componentId: "sc-1knyk5x-4"
})(["font-size:1.2rem;color:", ";margin-right:0.4rem;"], _color.red.z400);

AvatarCard.Wrapper = _styledComponents["default"].section.withConfig({
  displayName: "AvatarCard__Wrapper",
  componentId: "sc-1knyk5x-5"
})(["width:", ";display:flex;flex-direction:column;justify-content:center;"], getWidth);

var getWidth = function getWidth(props) {
  return (0, _lodashEs.get)(props, "width", "100%");
};

var _default = AvatarCard;
exports["default"] = _default;
