// webuikit/ContainerGrid/ContainerGrid.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var ContainerGrid = function ContainerGrid(props) {
  var _props$children = props.children,
      children = _props$children === void 0 ? null : _props$children,
      _props$gridGap = props.gridGap,
      gridGap = _props$gridGap === void 0 ? "" : _props$gridGap,
      _props$columns = props.columns,
      columns = _props$columns === void 0 ? "" : _props$columns,
      _props$mobileColumns = props.mobileColumns,
      mobileColumns = _props$mobileColumns === void 0 ? "" : _props$mobileColumns,
      _props$tabletColumns = props.tabletColumns,
      tabletColumns = _props$tabletColumns === void 0 ? "" : _props$tabletColumns;
  return _react["default"].createElement(ContainerGrid.Wrapper, {
    columns: getGrid(columns),
    mobileColumns: getGrid(mobileColumns),
    tabletColumns: getGrid(tabletColumns),
    gridGap: gridGap
  }, children);
};

var getGrid = function getGrid(grid) {
  var sections = grid.trim().split(" ");
  if (!sections.length) return "";
  return sections.map(function (sec) {
    return sec + "fr";
  }).join(" ");
};

ContainerGrid.propTypes = {
  children: _propTypes["default"].node,
  gridGap: _propTypes["default"].string,
  columns: _propTypes["default"].string,
  mobileColumns: _propTypes["default"].string,
  tabletColumns: _propTypes["default"].string
};
ContainerGrid.Item = _styledComponents["default"].div.withConfig({
  displayName: "ContainerGrid__Item",
  componentId: "h93c9m-0"
})(["", " ", " ", " display:flex;justify-content:center;"], function (props) {
  return !!props.gridColumn && (0, _styledComponents.css)(["grid-column:", ";"], props.gridColumn);
}, function (props) {
  return !!props.tabletGridColumn && (0, _styledComponents.css)(["@media (min-width:481px) and (max-width:768px){grid-column:", ";}"], props.tabletGridColumn);
}, function (props) {
  return !!props.mobileGridColumn && (0, _styledComponents.css)(["@media (max-width:480px){grid-column:", ";}"], props.mobileGridColumn);
});
ContainerGrid.Wrapper = _styledComponents["default"].section.withConfig({
  displayName: "ContainerGrid__Wrapper",
  componentId: "h93c9m-1"
})(["display:grid;max-width:90rem;align-items:flex-start;@media (min-width:481px) and (max-width:768px){width:100%;max-width:76.8rem;}", " grid-template-columns:", ";", " ", ""], function (props) {
  return !!props.gridGap && (0, _styledComponents.css)(["grid-gap:", ";"], props.gridGap);
}, function (props) {
  return props.columns;
}, function (props) {
  return !!props.mobileColumns && (0, _styledComponents.css)(["@media (max-width:480px){grid-template-columns:", ";}"], props.mobileColumns);
}, function (props) {
  return !!props.tabletColumns && (0, _styledComponents.css)(["@media (min-width:481px) and (max-width:768px){grid-template-columns:", ";}"], props.tabletColumns);
});
var _default = ContainerGrid;
exports["default"] = _default;
