// webuikit/Searchbox/Searchbox.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Icon = _interopRequireDefault(require("../Icon"));

var _Spinner = _interopRequireDefault(require("../Spinner"));

var _commonHelper = require("../helpers/commonHelper");

var _lodashEs = require("lodash-es");

var _OnClickOutside = _interopRequireDefault(require("../HOC/OnClickOutside"));

var _Textfield = _interopRequireDefault(require("./Textfield"));

var _misc = _interopRequireDefault(require("../tokens/misc"));

var _color = require("../tokens/color");

var Searchbox = function Searchbox(_ref) {
  var width = _ref.width,
      placeholder = _ref.placeholder,
      rightIcon = _ref.rightIcon,
      children = _ref.children,
      onChange = _ref.onChange,
      onSelect = _ref.onSelect,
      disabled = _ref.disabled,
      _ref$type = _ref.type,
      type = _ref$type === void 0 ? "" : _ref$type,
      _ref$tagType = _ref.tagType,
      tagType = _ref$tagType === void 0 ? "default" : _ref$tagType;

  var _useState = (0, _react.useState)(""),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      searchString = _useState2[0],
      setSearchString = _useState2[1];

  var _useState3 = (0, _react.useState)(false),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      loading = _useState4[0],
      setLoading = _useState4[1];

  var _useState5 = (0, _react.useState)(false),
      _useState6 = (0, _slicedToArray2["default"])(_useState5, 2),
      showMenu = _useState6[0],
      setMenuVisibility = _useState6[1];

  var _useState7 = (0, _react.useState)([]),
      _useState8 = (0, _slicedToArray2["default"])(_useState7, 2),
      tag = _useState8[0],
      setTag = _useState8[1];

  var _useState9 = (0, _react.useState)(null),
      _useState10 = (0, _slicedToArray2["default"])(_useState9, 2),
      childInputRef = _useState10[0],
      setChildInputRef = _useState10[1];

  var hasInput = !!searchString.length;
  var isTag = type === "tag";
  var node = (0, _react.useRef)();
  var parent = (0, _react.useRef)();

  var onInputChange = function onInputChange(e) {
    var value = e.target.value.trim() || "";
    setSearchString(e.target.value);

    if (value) {
      setLoading(true);
      setMenuVisibility(true);
      parent.current.debouncedOnChange(value);
    } else {
      setMenuVisibility(false);
    }
  };

  var handleRightIconClick = function handleRightIconClick(e) {
    e.stopPropagation();
    node.current.focus();
    setSearchString("");
    setMenuVisibility(false);
    setLoading(false);
  };

  var handleClick = function handleClick() {
    if (hasInput) setMenuVisibility(!showMenu);
  };

  var handleOnSelect = function handleOnSelect(value) {
    if (isTag) {
      setSearchString("");
      var newTags = [].concat((0, _toConsumableArray2["default"])(tag), [value]);
      setTag(newTags);
      onSelect(newTags);
    } else {
      setSearchString(value);
      onSelect(value);
    }
  };

  var onRemoveItem = function onRemoveItem(item) {
    var filtered = tag.filter(function (w) {
      return w !== item;
    });
    setTag(filtered);
    onSelect(filtered);
  };

  var suppressClick = function suppressClick(e) {
    if (showMenu) e.stopPropagation();
  };

  var handleFocus = function handleFocus() {
    isTag ? childInputRef.current.focus() : node.current.focus();
  };

  (0, _react.useEffect)(function () {
    if (children.length) setLoading(false);
    parent.current.debouncedOnChange = (0, _lodashEs.debounce)(onChange, 400);
  }, [children, onChange]);

  var closeMenu = function closeMenu() {
    return setMenuVisibility(false);
  };

  return _react["default"].createElement(_OnClickOutside["default"], {
    onClickOutside: closeMenu
  }, _react["default"].createElement(Searchbox.Wrapper, {
    width: width,
    onClick: handleClick,
    ref: parent,
    isTag: isTag,
    tabIndex: "0",
    "aria-labelledby": "searchbox-label"
  }, _react["default"].createElement(Searchbox.label, {
    id: "searchbox-label"
  }, placeholder), _react["default"].createElement(Searchbox.Header, {
    showMenu: showMenu,
    noPadding: isTag && tag.length,
    onClick: handleFocus
  }, _react["default"].createElement(Searchbox.Left, {
    isTag: isTag
  }, !isTag && _react["default"].createElement(_Icon["default"], {
    name: "search",
    size: 18,
    color: _color.grey.z400
  }), isTag ? _react["default"].createElement(_Textfield["default"], {
    placeholder: placeholder,
    searchStr: searchString,
    onChange: onInputChange,
    onRemove: onRemoveItem,
    value: searchString,
    tags: tag,
    type: tagType,
    setInputRef: setChildInputRef
  }) : _react["default"].createElement(Searchbox.Input, {
    placeholder: placeholder,
    onChange: onInputChange,
    value: searchString,
    ref: node,
    onClick: suppressClick,
    disabled: disabled,
    hasInput: hasInput
  })), !isTag && _react["default"].createElement(RightIcon, null, _react["default"].createElement(_Icon["default"], {
    onClick: handleRightIconClick,
    onKeyUp: (0, _commonHelper.onKeySelect)(handleRightIconClick),
    name: hasInput ? "cross" : rightIcon,
    size: 14,
    color: _color.grey.z400,
    showPointer: hasInput,
    "aria-label": hasInput ? "clear input" : "",
    tabIndex: "0"
  }))), _react["default"].createElement(Searchbox.Holder, null, _react["default"].createElement(Searchbox.ListWrapper, {
    isOpen: showMenu,
    tabIndex: "-1"
  }, showMenu && _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement(Searchbox.Cover, null), _react["default"].createElement(HR, null), loading && _react["default"].createElement(SpinnerHolder, null, _react["default"].createElement(_Spinner["default"], {
    color: _color.green.z500
  })), !loading && (0, _commonHelper.cloneChildren)(children, {
    _passClick: handleOnSelect,
    bold: searchString
  }))))));
};

Searchbox.propTypes = {
  width: _propTypes["default"].string,
  placeholder: _propTypes["default"].string,
  rightIcon: _propTypes["default"].string,
  children: _propTypes["default"].node,
  onChange: _propTypes["default"].func.isRequired,
  onSelect: _propTypes["default"].func.isRequired,
  disabled: _propTypes["default"].bool,
  type: _propTypes["default"].string,
  tagType: _propTypes["default"].oneOf(["red", "green", "default"])
};
Searchbox.defaultProps = {
  placeholder: "Start typing to search...",
  rightIcon: "downTriangle",
  disabled: false
};
Searchbox.label = _styledComponents["default"].label.withConfig({
  displayName: "Searchbox__label",
  componentId: "at2bon-0"
})(["display:none;"]);
Searchbox.Holder = _styledComponents["default"].section.withConfig({
  displayName: "Searchbox__Holder",
  componentId: "at2bon-1"
})(["position:relative;"]);

var RightIcon = _styledComponents["default"].span.withConfig({
  displayName: "Searchbox__RightIcon",
  componentId: "at2bon-2"
})(["margin-left:3rem;"]);

var SpinnerHolder = _styledComponents["default"].div.withConfig({
  displayName: "Searchbox__SpinnerHolder",
  componentId: "at2bon-3"
})(["margin:1.5rem 0 0.5rem 0;"]);

var HR = _styledComponents["default"].hr.withConfig({
  displayName: "Searchbox__HR",
  componentId: "at2bon-4"
})(["border-left:0;border-top:", ";margin:2px 20px 0 20px;"], "1px solid " + _color.grey.z200);

Searchbox.Cover = _styledComponents["default"].div.withConfig({
  displayName: "Searchbox__Cover",
  componentId: "at2bon-5"
})(["position:absolute;top:-6px;left:0;width:100%;background:#fff;height:8px;"]);
Searchbox.ListWrapper = _styledComponents["default"].section.withConfig({
  displayName: "Searchbox__ListWrapper",
  componentId: "at2bon-6"
})(["width:100%;position:absolute;background:#fff;box-shadow:", ";border-radius:", ";border-top:none;text-align:center;z-index:10;padding-bottom:1rem;opacity:", ";transition:all ", " ease;"], function (props) {
  return props.isOpen ? "0px 2px 8px rgba(28, 28, 28, 0.15)" : "none";
}, "0 0 " + _misc["default"].radius + " " + _misc["default"].radius, function (props) {
  return props.isOpen ? "1" : "0";
}, _misc["default"].animationDuration);
Searchbox.Wrapper = _styledComponents["default"].div.withConfig({
  displayName: "Searchbox__Wrapper",
  componentId: "at2bon-7"
})(["position:relative;cursor:text;width:", ";"], function (props) {
  return getWidth(props);
});
Searchbox.Header = _styledComponents["default"].div.withConfig({
  displayName: "Searchbox__Header",
  componentId: "at2bon-8"
})(["position:relative;display:flex;align-items:center;justify-content:space-between;padding:", ";font-size:18px;border:", ";border-radius:", ";box-shadow:", ";:focus-within{box-shadow:0px 2px 8px rgba(28,28,28,0.15);}"], function (props) {
  return props.noPadding ? "0 0.6rem" : "1.55rem 2rem";
}, function (props) {
  return props.showMenu ? "0.5px solid transparent" : "0.5px solid " + _color.grey.z300;
}, function (props) {
  return props.showMenu ? _misc["default"].radius + " " + _misc["default"].radius + " 0 0" : _misc["default"].radius;
}, function (props) {
  return props.showMenu ? "0px 2px 8px rgba(28, 28, 28, 0.15)" : "none";
});
Searchbox.Left = _styledComponents["default"].div.withConfig({
  displayName: "Searchbox__Left",
  componentId: "at2bon-9"
})(["display:", ";align-items:center;width:", ";overflow:", ";white-space:", ";"], function (props) {
  return props.isTag ? "unset" : "flex";
}, function (props) {
  return props.isTag ? "100%" : "max-content";
}, function (props) {
  return props.isTag ? "inherit" : "hidden";
}, function (props) {
  return props.isTag ? "inherit" : "nowrap";
});
Searchbox.Input = _styledComponents["default"].input.withConfig({
  displayName: "Searchbox__Input",
  componentId: "at2bon-10"
})(["margin-left:0.5rem;margin-top:0.2rem;color:", ";overflow:hidden;text-overflow:ellipsis;outline:none;border:none;font-family:inherit;font-size:1.8rem;&::placeholder{color:", ";font-weight:300;}"], _color.black, _color.grey.z400);

var getWidth = function getWidth(props) {
  return props.width ? props.width : props.isTag ? "30rem" : "32rem";
};

var _default = Searchbox;
exports["default"] = _default;
