// webuikit/esm/ImageCard/ImageCard.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _lodashEs = require("lodash-es");

var _ZImage = _interopRequireDefault(require("../ZImage"));

var ImageCard = function ImageCard(props) {
  var _props$children = props.children,
      children = _props$children === void 0 ? null : _props$children,
      _props$grid = props.grid,
      grid = _props$grid === void 0 ? "" : _props$grid,
      _props$gridGap = props.gridGap,
      gridGap = _props$gridGap === void 0 ? "1rem" : _props$gridGap;

  var isReversed = function isReversed() {
    var first = _react["default"].Children.toArray(children)[0];

    return (0, _lodashEs.get)(first, "type.displayName") !== "ImageCard__Image";
  };

  return _react["default"].createElement(ImageCard.Wrapper, {
    grid: getGrid(grid),
    gridGap: gridGap,
    reversed: isReversed()
  }, children);
};

var getGrid = function getGrid(grid) {
  var sections = grid.trim().split(" ");
  if (!sections.length) return "";
  return sections.map(function (sec) {
    return sec + "fr";
  }).join(" ");
};

ImageCard.propTypes = {
  children: _propTypes["default"].node,
  grid: _propTypes["default"].string,
  gridGap: _propTypes["default"].string,
  pictureAlign: _propTypes["default"].oneOf(["left", "right"])
};
ImageCard.Wrapper = _styledComponents["default"].section.withConfig({
  displayName: "ImageCard__Wrapper",
  componentId: "sc-1qty564-0"
})(["align-items:center;", " @media (max-width:480px){display:flex;flex-direction:", ";}"], function (props) {
  return props.grid && (0, _styledComponents.css)(["display:grid;grid-template-columns:", ";", ""], props.grid, props.gridGap && (0, _styledComponents.css)(["grid-gap:", ";"], props.gridGap));
}, function (props) {
  return props.reversed ? "column-reverse" : "column";
});
ImageCard.Description = _styledComponents["default"].div.withConfig({
  displayName: "ImageCard__Description",
  componentId: "sc-1qty564-1"
})(["display:flex;flex-direction:column;@media (max-width:480px){text-align:center;}"]);
ImageCard.Image = (0, _styledComponents["default"])(_ZImage["default"]).withConfig({
  displayName: "ImageCard__Image",
  componentId: "sc-1qty564-2"
})(["max-width:450px;width:100%;@media (max-width:480px){margin-bottom:5rem;position:relative;float:left;}"]);
var _default = ImageCard;
exports["default"] = _default;
