// webuikit/es/RestaurantGeneralCard/generalCardHelpers.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getHeadlineColor = exports.getState = exports.renderBtn = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _Button = _interopRequireDefault(require("../Button"));

var _constants = require("./constants");

var _color = require("../tokens/color");

var _disabledState, _activeState;

var disabledState = (_disabledState = {}, (0, _defineProperty2["default"])(_disabledState, _constants.IMG_FILTER, "grayscale(100%)"), (0, _defineProperty2["default"])(_disabledState, _constants.HEADING_COLOR, _color.grey.z500), (0, _defineProperty2["default"])(_disabledState, _constants.TYPE_COLOR, _color.grey.z400), (0, _defineProperty2["default"])(_disabledState, _constants.RATING_COLOR, _color.grey.z300), (0, _defineProperty2["default"])(_disabledState, _constants.INFO_COLOR, _color.grey.z400), _disabledState);
var activeState = (_activeState = {}, (0, _defineProperty2["default"])(_activeState, _constants.IMG_FILTER, "none"), (0, _defineProperty2["default"])(_activeState, _constants.HEADING_COLOR, _color.black), (0, _defineProperty2["default"])(_activeState, _constants.TYPE_COLOR, _color.grey.z800), (0, _defineProperty2["default"])(_activeState, _constants.RATING_COLOR, _color.green.z500), (0, _defineProperty2["default"])(_activeState, _constants.INFO_COLOR, _color.grey.z600), _activeState);

var renderBtn = function renderBtn(label, disabled) {
  return _react["default"].createElement(_Button["default"], {
    btnColor: "red",
    size: "small",
    appearance: "outline",
    icon: "right-triangle",
    iconAlign: "right",
    disabled: disabled
  }, label);
};

exports.renderBtn = renderBtn;

var getHeadlineColor = function getHeadlineColor(_ref) {
  var color = _ref.color;
  return color === "blue" ? _color.blue.z500 : _color.red.z500;
};

exports.getHeadlineColor = getHeadlineColor;

var getState = function getState(key) {
  return function (_ref2) {
    var disabled = _ref2.disabled;
    return disabled ? disabledState[key] : activeState[key];
  };
};

exports.getState = getState;
