// webuikit/es/RestaurantGeneralCard/RestaurantGeneralCard.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _ZImage = _interopRequireDefault(require("../ZImage"));

var _constants = require("./constants");

var _generalCardHelpers = require("./generalCardHelpers");

// Libraries
// Components
// Constants
// Helpers
var RestaurantGeneralCard = function RestaurantGeneralCard(props) {
  return _react["default"].createElement(RestaurantGeneralCardWrapper, null, _react["default"].createElement(Img, {
    disabled: props.disabled,
    src: props.imageUrl
  }), _react["default"].createElement(Details, null, _react["default"].createElement(Info, null, _react["default"].createElement("article", null, _react["default"].createElement(Title, {
    color: props.headlineColor
  }, props.headline), _react["default"].createElement(Heading, {
    disabled: props.disabled
  }, props.resName), _react["default"].createElement(Type, {
    disabled: props.disabled
  }, props.resType)), _react["default"].createElement("article", null, _react["default"].createElement(InfoType, {
    disabled: props.disabled
  }, props.distance), _react["default"].createElement(Separator, null), _react["default"].createElement(InfoType, {
    disabled: props.disabled
  }, props.location), _react["default"].createElement(NType, {
    disabled: props.disabled
  }, props.pricing), _react["default"].createElement(ButtonWrapper, null, (0, _generalCardHelpers.renderBtn)(props.orderBtnLabel, props.disabled || !props.isOrderActive), _react["default"].createElement(BtnMargin, null), (0, _generalCardHelpers.renderBtn)(props.bookBtnLabel, props.disabled || !props.isBookingActive)))), _react["default"].createElement(Rating, {
    disabled: props.disabled
  }, props.rating)));
};

var RestaurantGeneralCardWrapper = _styledComponents["default"].section.withConfig({
  displayName: "RestaurantGeneralCard__RestaurantGeneralCardWrapper",
  componentId: "sc-2y61hc-0"
})(["display:flex;"]);

var ButtonWrapper = _styledComponents["default"].section.withConfig({
  displayName: "RestaurantGeneralCard__ButtonWrapper",
  componentId: "sc-2y61hc-1"
})(["margin-top:0.5rem;"]);

var BtnMargin = _styledComponents["default"].span.withConfig({
  displayName: "RestaurantGeneralCard__BtnMargin",
  componentId: "sc-2y61hc-2"
})(["margin-left:1rem;"]);

var Title = _styledComponents["default"].article.withConfig({
  displayName: "RestaurantGeneralCard__Title",
  componentId: "sc-2y61hc-3"
})(["font-size:1.4rem;line-height:2.1rem;color:", ";margin-top:0.7rem;"], _generalCardHelpers.getHeadlineColor);

var Img = (0, _styledComponents["default"])(_ZImage["default"]).withConfig({
  displayName: "RestaurantGeneralCard__Img",
  componentId: "sc-2y61hc-4"
})(["height:19.4rem;width:23rem;border-top-left-radius:0.6rem;border-bottom-left-radius:0.6rem;filter:", ";"], (0, _generalCardHelpers.getState)(_constants.IMG_FILTER));

var Heading = _styledComponents["default"].article.withConfig({
  displayName: "RestaurantGeneralCard__Heading",
  componentId: "sc-2y61hc-5"
})(["font-weight:400;font-size:2rem;line-height:2.4rem;color:", ";"], (0, _generalCardHelpers.getState)(_constants.HEADING_COLOR));

var InfoType = _styledComponents["default"].span.withConfig({
  displayName: "RestaurantGeneralCard__InfoType",
  componentId: "sc-2y61hc-6"
})(["font-size:1.4rem;line-height:2.1rem;color:", ";"], (0, _generalCardHelpers.getState)(_constants.INFO_COLOR));

var Details = _styledComponents["default"].section.withConfig({
  displayName: "RestaurantGeneralCard__Details",
  componentId: "sc-2y61hc-7"
})(["display:flex;width:40rem;padding:0 1.6rem 1.2rem;border-top:1px solid #e8e8e8;border-right:1px solid #e8e8e8;border-bottom:1px solid #e8e8e8;border-top-right-radius:0.6rem;border-bottom-right-radius:0.6rem;"]);

var Info = _styledComponents["default"].section.withConfig({
  displayName: "RestaurantGeneralCard__Info",
  componentId: "sc-2y61hc-8"
})(["display:flex;flex-direction:column;justify-content:space-between;height:100%;flex-grow:1;padding-right:2rem;"]);

var Rating = _styledComponents["default"].article.withConfig({
  displayName: "RestaurantGeneralCard__Rating",
  componentId: "sc-2y61hc-9"
})(["background-color:", ";border-radius:0.6rem;font-size:2rem;line-height:2.9rem;padding:0.1rem 0.9rem;height:fit-content;color:#fff;margin-top:1.1rem;"], (0, _generalCardHelpers.getState)(_constants.RATING_COLOR));

var Type = _styledComponents["default"].span.withConfig({
  displayName: "RestaurantGeneralCard__Type",
  componentId: "sc-2y61hc-10"
})(["color:", ";font-size:1.4rem;line-height:2.1rem;"], (0, _generalCardHelpers.getState)(_constants.TYPE_COLOR));

var NType = _styledComponents["default"].section.withConfig({
  displayName: "RestaurantGeneralCard__NType",
  componentId: "sc-2y61hc-11"
})(["color:", ";padding-top:0.3rem;font-size:1.4rem;line-height:2.1rem;"], (0, _generalCardHelpers.getState)(_constants.TYPE_COLOR));

var Separator = _styledComponents["default"].span.withConfig({
  displayName: "RestaurantGeneralCard__Separator",
  componentId: "sc-2y61hc-12"
})(["width:1px;background-color:#cfcfcf;height:100%;border:0.5px solid #cfcfcf;margin:0 0.9rem;border-radius:2px;"]);

RestaurantGeneralCard.propTypes = {
  imageUrl: _propTypes["default"].string,
  resName: _propTypes["default"].string.isRequired,
  resType: _propTypes["default"].string.isRequired,
  distance: _propTypes["default"].string.isRequired,
  location: _propTypes["default"].string.isRequired,
  rating: _propTypes["default"].string.isRequired,
  pricing: _propTypes["default"].string.isRequired,
  disabled: _propTypes["default"].bool,
  orderBtnLabel: _propTypes["default"].string.isRequired,
  bookBtnLabel: _propTypes["default"].string.isRequired,
  isOrderActive: _propTypes["default"].bool,
  isBookingActive: _propTypes["default"].bool,
  headline: _propTypes["default"].string.isRequired,
  headlineColor: _propTypes["default"].oneOf(["blue", "red"])
};
RestaurantGeneralCard.defaultProps = {
  disabled: false,
  isOrderActive: true,
  isBookingActive: true
};
var _default = RestaurantGeneralCard;
exports["default"] = _default;
