// webuikit/Spinner/Spinner.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.SpinnerCircular = exports.spinnerTypes = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _color = require("../tokens/color");

var _commonHelper = require("../helpers/commonHelper");

var loadingKeyframe = (0, _styledComponents.keyframes)(["0%,80%,100%{transform:scale(0);}40%{transform:scale(1.0);}"]);
var spinnerTypes = {
  small: {
    width: 1
  },
  medium: {
    width: 1.8
  },
  large: {
    width: 2.4
  }
};
exports.spinnerTypes = spinnerTypes;

var Spinner = function Spinner(props) {
  var variant = spinnerTypes[props.size];
  return _react["default"].createElement(Spinner.container, {
    variant: variant
  }, _react["default"].createElement(Spinner.dot, (0, _extends2["default"])({}, props, {
    variant: variant,
    className: "bounce1"
  })), _react["default"].createElement(Spinner.dot, (0, _extends2["default"])({}, props, {
    className: "bounce2",
    variant: variant
  })), _react["default"].createElement(Spinner.dot, (0, _extends2["default"])({}, props, {
    className: "bounce3",
    variant: variant
  })));
};

var SpinnerCircular = function SpinnerCircular(props) {
  var _props$size = props.size,
      size = _props$size === void 0 ? "small" : _props$size;
  var variant = spinnerTypes[size];
  return _react["default"].createElement(SpinnerCircular.container, {
    width: variant.width,
    color: props.color
  });
};

exports.SpinnerCircular = SpinnerCircular;
SpinnerCircular.propTypes = {
  size: _propTypes["default"].oneOf(["small", "medium", "large"]),
  color: _propTypes["default"].string
};
var circularSpin = (0, _styledComponents.keyframes)(["0%{transform:rotate(0deg);}100%{transform:rotate(360deg);}"]);

var circularSpinnerBG = function circularSpinnerBG(props) {
  var color = props.color || _color.grey.z500;
  var rgb = (0, _commonHelper.hexToRgb)(color);
  return "rgb(".concat(rgb.r, ", ").concat(rgb.g, ", ").concat(rgb.b, ", 0.1)");
};

SpinnerCircular.container = _styledComponents["default"].div.withConfig({
  displayName: "Spinner__container",
  componentId: "sc-1d6aql7-0"
})(["width:", "rem;height:", "rem;border-radius:50%;border:", "rem solid ", ";border-top:", "rem solid ", ";animation:", " 1s infinite linear;"], function (props) {
  return 1.4 * props.width;
}, function (props) {
  return 1.4 * props.width;
}, function (props) {
  return 0.5 / 2 * props.width;
}, circularSpinnerBG, function (props) {
  return 0.5 / 2 * props.width;
}, function (props) {
  return props.color;
}, circularSpin);
Spinner.container = _styledComponents["default"].div.withConfig({
  displayName: "Spinner__container",
  componentId: "sc-1d6aql7-1"
})(["display:inline-flex;vertical-align:middle;align-items:center;justify-content:center;width:", "rem;text-align:center;"], function (props) {
  return 4 * props.variant.width;
});
Spinner.dot = _styledComponents["default"].div.withConfig({
  displayName: "Spinner__dot",
  componentId: "sc-1d6aql7-2"
})(["width:", "rem;height:", "rem;background-color:", ";border-radius:100%;margin-right:0.2rem;display:inline-block;-webkit-animation:", " 1.4s infinite ease-in-out both;animation:", " 1.4s infinite ease-in-out both;&.bounce1{-webkit-animation-delay:-0.32s;animation-delay:-0.32s;}&.bounce2{-webkit-animation-delay:-0.16s;animation-delay:-0.16s;}&.bounce3{}"], function (props) {
  return props.variant.width;
}, function (props) {
  return props.variant.width;
}, function (props) {
  return props.color;
}, loadingKeyframe, loadingKeyframe);
Spinner.propTypes = {
  inverse: _propTypes["default"].bool,

  /** Color could be a hex code or a string*/
  color: _propTypes["default"].string,
  size: _propTypes["default"].oneOf(["small", "medium", "large"])
};
Spinner.defaultProps = {
  inverse: false,
  size: "small",
  color: "rgba(0,0,0)"
};
var _default = Spinner;
exports["default"] = _default;
