// webuikit/Switch/RadioSwitch.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _misc = _interopRequireDefault(require("../tokens/misc"));

var _lodashEs = require("lodash-es");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _color = require("../tokens/color");

var _Icon = _interopRequireDefault(require("../Icon"));

var size = 26;
var spacing = 2;

var RadioSwitch = function RadioSwitch(_ref) {
  var onToggle = _ref.onToggle,
      readOnly = _ref.readOnly,
      active = _ref.active;

  var _useState = (0, _react.useState)(active),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      activeState = _useState2[0],
      toggleActiveState = _useState2[1];

  return _react["default"].createElement(RadioSwitch.wrapper, {
    onClick: function onClick() {
      onToggle(!activeState);
      return toggleActiveState(!activeState);
    }
  }, _react["default"].createElement(RadioSwitch.iconSpan, {
    on: activeState && !readOnly
  }, _react["default"].createElement(_Icon["default"], {
    color: _color.red.z500,
    size: size,
    name: "checkCircle"
  })), _react["default"].createElement(RadioSwitch.inactiveSpan, {
    on: !activeState || readOnly
  }));
};

RadioSwitch.wrapper = _styledComponents["default"].span.withConfig({
  displayName: "RadioSwitch__wrapper",
  componentId: "sc-1p27j6w-0"
})(["display:inline-block;position:relative;cursor:pointer;width:", "rem;height:", "rem;appearance:none;outline:none;"], size / 10, size / 10);
RadioSwitch.inactiveSpan = _styledComponents["default"].span.withConfig({
  displayName: "RadioSwitch__inactiveSpan",
  componentId: "sc-1p27j6w-1"
})(["display:inline-block;position:absolute;appearance:none;top:0;left:0;width:100%;height:100%;border:", "rem ", " solid;border-radius:50%;transition:opacity ", ";opacity:", ";"], spacing / 10, _color.grey.z400, _misc["default"].shortAnimDuration, function (props) {
  return props.on ? "1" : 0;
});
RadioSwitch.iconSpan = _styledComponents["default"].span.withConfig({
  displayName: "RadioSwitch__iconSpan",
  componentId: "sc-1p27j6w-2"
})(["display:inline-block;position:absolute;top:0;left:0;width:", "rem;appearance:none;height:", "rem;transition:opacity ", ";opacity:", ";"], size / 10, size / 10, _misc["default"].shortAnimDuration, function (props) {
  return props.on ? "1" : 0;
});
RadioSwitch.propTypes = {
  readOnly: _propTypes["default"].bool,
  size: _propTypes["default"].number,
  spacing: _propTypes["default"].number,
  onToggle: _propTypes["default"].func,
  active: _propTypes["default"].bool
};
RadioSwitch.defaultProps = {
  readOnly: false,
  size: 26,
  spacing: 2,
  onToggle: _lodashEs.noop
};
var _default = RadioSwitch;
exports["default"] = _default;
