// webuikit/es/ZImage/ZImage.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _color = require("../tokens/color");

var ZImage = function ZImage(props) {
  var _props$src = props.src,
      src = _props$src === void 0 ? "" : _props$src,
      _props$base = props.base64,
      base64 = _props$base === void 0 ? "" : _props$base,
      _props$ratio = props.ratio,
      ratio = _props$ratio === void 0 ? 0 : _props$ratio,
      _props$container = props.container,
      container = _props$container === void 0 ? null : _props$container,
      _props$alt = props.alt,
      alt = _props$alt === void 0 ? "image" : _props$alt;
  var ref = (0, _react.useRef)();

  var _useState = (0, _react.useState)(""),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      imgSrc = _useState2[0],
      setSrc = _useState2[1];

  var _useState3 = (0, _react.useState)(false),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      isImageLoaded = _useState4[0],
      setLoaded = _useState4[1];

  var handleOnImageLoad = function handleOnImageLoad() {
    return setLoaded(true);
  };

  (0, _react.useEffect)(function () {
    var intersected = function intersected(_ref) {
      var _ref2 = (0, _slicedToArray2["default"])(_ref, 1),
          entry = _ref2[0];

      var _entry$isIntersecting = entry.isIntersecting,
          isIntersecting = _entry$isIntersecting === void 0 ? false : _entry$isIntersecting;
      if (isIntersecting && imgSrc !== src) setSrc(src);
    };

    var options = {
      root: container,
      threshold: ratio
    };
    var observer = new IntersectionObserver(intersected, options);
    var currentNode = ref.current;
    if (currentNode) observer.observe(currentNode);
    return function () {
      return observer.unobserve(currentNode);
    };
  }, []);
  return _react["default"].createElement(ZImage.Container, (0, _extends2["default"])({
    ref: ref
  }, props), !base64 ? _react["default"].createElement(ZImage.Shimmer, {
    isLoaded: !!imgSrc && isImageLoaded
  }) : _react["default"].createElement(ZImage.base64, {
    isLoaded: !!imgSrc && isImageLoaded,
    src: base64
  }), _react["default"].createElement(ZImage.img, {
    alt: alt,
    src: imgSrc,
    isLoaded: !!imgSrc && isImageLoaded,
    onLoad: handleOnImageLoad,
    loading: "lazy" // More info on Native lazy load :  https://addyosmani.com/blog/lazy-loading/

  }));
};

ZImage.propTypes = {
  src: _propTypes["default"].string,
  base64: _propTypes["default"].string,
  ratio: _propTypes["default"].number,
  container: _propTypes["default"].node,
  alt: _propTypes["default"].string.isRequired
};
ZImage.Container = _styledComponents["default"].div.withConfig({
  displayName: "ZImage__Container",
  componentId: "sc-1twaxvn-0"
})(["position:relative;max-width:100%;width:", ";height:", ";max-height:", ";overflow:hidden;"], function (props) {
  return props.width;
}, function (props) {
  return props.height ? props.height : "auto";
}, function (props) {
  return props.maxHeight;
});
var shimmerAnimation = (0, _styledComponents.keyframes)(["0%{background-position:-50rem 0;}100%{background-position:50rem 0;}"]);
ZImage.base64 = _styledComponents["default"].img.withConfig({
  displayName: "ZImage__base64",
  componentId: "sc-1twaxvn-1"
})(["position:absolute;top:0;left:0;width:100%;height:100%;opacity:", ";will-change:transform,opacity;transition:opacity 0.5s ease-in;"], function (props) {
  return props.isLoaded ? 0 : 1;
});
ZImage.Shimmer = _styledComponents["default"].div.withConfig({
  displayName: "ZImage__Shimmer",
  componentId: "sc-1twaxvn-2"
})(["width:100%;position:absolute;top:0;left:0;height:100%;background:", ";background-image:linear-gradient( to right,", " 0%,", " 20%,", " 40%,", " 100% );background-repeat:no-repeat;opacity:", ";transition:opacity 0.5s ease-out;animation:", " 1s infinite linear forwards;"], _color.grey.z95, _color.grey.z95, _color.grey.z100, _color.grey.z95, _color.grey.z95, function (props) {
  return props.isLoaded ? 0 : 1;
}, shimmerAnimation);
ZImage.img = _styledComponents["default"].img.withConfig({
  displayName: "ZImage__img",
  componentId: "sc-1twaxvn-3"
})(["width:100%;height:100%;object-fit:cover;opacity:", ";will-change:transform,opacity;transition:opacity 0.5s ease-in;"], function (props) {
  return props.isLoaded ? 1 : 0;
});
var _default = ZImage;
exports["default"] = _default;
