// webuikit/Layouts/Header.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _color = require("../tokens/color");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Navbar = _interopRequireWildcard(require("./Navbar"));

var _uniqueIdHelper = _interopRequireDefault(require("../helpers/uniqueIdHelper"));

var navId = (0, _uniqueIdHelper["default"])("navigation");
var headerHeight = "7.2rem";
var maxContentWidth = "90rem";

var Header = function Header(props) {
  var logoImg = props.logoImg;

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      isMenuOpened = _useState2[0],
      toggleMenuOpen = _useState2[1];

  var children = _react["default"].Children.map(props.children, function (child, index) {
    return _react["default"].cloneElement(child, {
      index: index,
      open: isMenuOpened,
      navId: navId
    });
  });

  var toggleMobileMenu = function toggleMobileMenu() {
    var menuState = !isMenuOpened;

    if (menuState) {
      document.body.style.overflow = "hidden !important";
      document.body.style.height = "100% !important";
    } else {
      document.body.style.overflow = "";
      document.body.style.height = "";
    }

    toggleMenuOpen(menuState);
  };

  return _react["default"].createElement(Header.wrapper, null, _react["default"].createElement(Header.Element, null, _react["default"].createElement(MobileHeader, null, _react["default"].createElement(MobileHeader.Ele, null, _react["default"].createElement(_Navbar.MenuIcon, {
    "aria-label": "Menu",
    "aria-controls": navId,
    "aria-expanded": isMenuOpened,
    role: "button",
    open: isMenuOpened,
    onClick: toggleMobileMenu
  })), _react["default"].createElement(MobileHeader.Ele, null, _react["default"].createElement(Header.mobileLogo, {
    src: logoImg
  })), _react["default"].createElement(MobileHeader.Ele, null), " "), _react["default"].createElement(Header.Logo, {
    src: logoImg
  }), children));
};

Header.wrapper = _styledComponents["default"].header.withConfig({
  displayName: "Header__wrapper",
  componentId: "sc-9em35a-0"
})(["width:100%;max-width:100%;display:block;height:", ";justify-content:space-between;align-items:center;background-color:", ";"], headerHeight, _color.white);
Header.Element = _styledComponents["default"].nav.withConfig({
  displayName: "Header__Element",
  componentId: "sc-9em35a-1"
})(["width:100%;max-width:", ";margin:0 auto;display:flex;flex-wrap:nowrap;height:100%;justify-content:space-between;align-items:center;position:relative;@media screen and (max-width:", "){&{}}"], maxContentWidth, maxContentWidth);

var MobileHeader = _styledComponents["default"].ul.withConfig({
  displayName: "Header__MobileHeader",
  componentId: "sc-9em35a-2"
})(["display:none;height:", ";overflow:hidden;position:absolute;width:100%;margin:0;padding:0 1.5rem;@media screen and (max-width:768px){&{display:flex;justify-content:space-between;}}"], headerHeight);

MobileHeader.Ele = _styledComponents["default"].li.withConfig({
  displayName: "Header__Ele",
  componentId: "sc-9em35a-3"
})(["display:none;z-index:2;height:100%;width:auto;display:flex;border-bottom:none;align-items:center;justify-content:center;"]);
Header.Logo = _styledComponents["default"].img.withConfig({
  displayName: "Header__Logo",
  componentId: "sc-9em35a-4"
})(["max-width:14rem;height:2.7rem;display:block;@media screen and (max-width:768px){&{display:none;}}"]);
Header.mobileLogo = (0, _styledComponents["default"])(Header.Logo).withConfig({
  displayName: "Header__mobileLogo",
  componentId: "sc-9em35a-5"
})(["display:none;@media screen and (max-width:768px){&{display:block;}}"]);
Header.propTypes = {
  children: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].node]).isRequired,
  logoImg: _propTypes["default"].string.isRequired
};
var _default = Header;
exports["default"] = _default;
