// webuikit/Layouts/Navbar.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MenuIcon = exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _color = require("../tokens/color");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _typography = require("../tokens/typography");

var _uniqueIdHelper = _interopRequireDefault(require("../helpers/uniqueIdHelper"));

/* eslint-disable indent */
var navId = (0, _uniqueIdHelper["default"])("navigation");

var Navbar = function Navbar(props) {
  var children = props.children,
      open = props.open,
      navId = props.navId;
  return (// <Navbar.Wrapper>
    _react["default"].createElement(Navbar.Element, {
      id: navId,
      role: "menu",
      "aria-label": "navbar",
      open: open,
      "aria-expanded": open
    }, children.map(function (child, index) {
      return child.props.mobileLink && child.props.children ? _react["default"].createElement(Navbar.item, {
        key: index,
        mobileLink: child.props.mobileLink
      }, child.props.children) : _react["default"].createElement(Navbar.item, {
        key: index
      }, child);
    })) // </Navbar.Wrapper>

  );
};

var MenuIcon = function MenuIcon(props) {
  return _react["default"].createElement(MenuIcon.wrapper, props, _react["default"].createElement(MenuIcon.span, null), _react["default"].createElement(MenuIcon.span, null), _react["default"].createElement(MenuIcon.span, null));
};

exports.MenuIcon = MenuIcon;
MenuIcon.wrapper = _styledComponents["default"].div.withConfig({
  displayName: "Navbar__wrapper",
  componentId: "sc-1n8gxvy-0"
})(["width:2rem;height:1.5rem;position:relative;transform:rotate(0deg);transition:transform 0.15s ease-in-out;cursor:pointer;opacity:0;@media screen and (max-width:768px){&{opacity:1;}", "}"], function (props) {
  return props.open && (0, _styledComponents.css)(["& > span:nth-child(1){transform:rotate(45deg);left:0.2rem;top:-0.1rem;}& > span:nth-child(2){width:0%;opacity:0;}& > span:nth-child(3){transform:rotate(-45deg);left:0.2rem;top:1.3rem;}"]);
});
MenuIcon.span = _styledComponents["default"].span.withConfig({
  displayName: "Navbar__span",
  componentId: "sc-1n8gxvy-1"
})(["transform-origin:left center;display:block;position:absolute;cursor:pointer;height:3px;width:100%;background:", ";border-radius:0;opacity:1;left:0;transform:rotate(0deg);transition:transform 0.15s ease-in-out,all 0.15s linear;&:nth-child(1){top:0;}&:nth-child(2){top:0.6rem;}&:nth-child(3){top:1.2rem;}"], _color.black);
var activeLinkStyle = (0, _styledComponents.css)(["color:", ";border-bottom-color:", ";"], _color.black, _color.red.z500);
Navbar.Wrapper = _styledComponents["default"].div.withConfig({
  displayName: "Navbar__Wrapper",
  componentId: "sc-1n8gxvy-2"
})(["display:flex;align-items:center;flex-grow:1;justify-content:flex-end;@media screen and (max-width:768px){&{position:absolute;width:100%;height:100%;left:0;}}"]);
Navbar.Element = _styledComponents["default"].ul.withConfig({
  displayName: "Navbar__Element",
  componentId: "sc-1n8gxvy-3"
})(["width:100%;list-style:none;padding:0;max-width:100%;height:72px;padding-top:1.5rem;padding-bottom:1.7rem;display:flex;flex-wrap:nowrap;background:transparent;align-items:center;display:flex;-webkit-box-pack:justify;-ms-flex-pack:justify;justify-content:flex-end;user-select:none;cursor:pointer;opacity:1;transition-delay:0s;padding:1.7rem 0;transition:opacity 0.25s linear;height 0.56s cubic-bezier(0.52,0.16,0.24,1);> *{margin-right:3.2rem;}> *:last-child{margin-right:0;}@media screen and (max-width:768px){&{opacity:0;visibility:hidden;flex-direction:column;position:absolute;width:100%;align-items:flex-start;justify-content:flex-start;background:", ";top:100%;left:0;padding:10px 22px;height:100vh;", "}> *{margin-right:0;}}"], _color.white, function (props) {
  return props.open && (0, _styledComponents.css)(["opacity:1;visibility:visible;"]);
});
Navbar.item = _styledComponents["default"].li.withConfig({
  displayName: "Navbar__item",
  componentId: "sc-1n8gxvy-4"
})(["> *{margin-right:3.2rem;}> *:last-child{margin-right:0;}@media screen and (max-width:768px){&{margin-bottom:3.5rem;width:100%;}&:nth-last-child{margin-bottom:0;}> *{margin-right:0;}", "}"], function (props) {
  return props.mobileLink && (0, _styledComponents.css)(["flex-direction:column;align-items:center;display:flex;align-self:flex-end;& > *{margin-bottom:2rem;}"]);
});
Navbar.propTypes = {
  children: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].node]).isRequired
};
Navbar.Link = _styledComponents["default"].a.withConfig({
  displayName: "Navbar__Link",
  componentId: "sc-1n8gxvy-5"
})(["font-size:", ";line-height:", ";text-align:right;vertical-align:middle;font-weight:", ";color:", ";padding:1rem 0;box-sizing:border-box;border-width:0;border-style:solid;border-color:transparent;border-bottom-width:0.2rem;&:hover{color:", ";}&:active,&:hover:active{", "}", ""], _typography.fontSizes.z400, _typography.lineHeight.medium, _typography.fontWeights.light, _color.grey.z700, _color.black, activeLinkStyle, function (props) {
  return props.active && activeLinkStyle;
});
Navbar.mobileLinks = _styledComponents["default"].aside.withConfig({
  displayName: "Navbar__mobileLinks",
  componentId: "sc-1n8gxvy-6"
})(["display:flex;@media screen and (max-width:768px){&{flex-direction:column;align-self:flex-end;}}"]);
var _default = Navbar;
exports["default"] = _default;
