// webuikit/Dropdown/Dropdown.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Icon = _interopRequireDefault(require("../Icon"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _color = require("../tokens/color");

var _commonHelper = require("../helpers/commonHelper");

var _OnClickOutside = _interopRequireDefault(require("../HOC/OnClickOutside"));

var _misc = _interopRequireDefault(require("../tokens/misc"));

var Dropdown = function Dropdown(props) {
  var value = props.value,
      width = props.width,
      leftIcon = props.leftIcon,
      rightIcon = props.rightIcon,
      onSelect = props.onSelect,
      children = props.children,
      _props$helperText = props.helperText,
      helperText = _props$helperText === void 0 ? "" : _props$helperText,
      _props$errorText = props.errorText,
      errorText = _props$errorText === void 0 ? "" : _props$errorText,
      _props$showArrow = props.showArrow,
      showArrow = _props$showArrow === void 0 ? false : _props$showArrow;

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      isMenuOpen = _useState2[0],
      toggleMenu = _useState2[1];

  var _useState3 = (0, _react.useState)(value),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      label = _useState4[0],
      setLabel = _useState4[1];

  var toggle = function toggle() {
    return toggleMenu(!isMenuOpen);
  };

  var closeMenu = function closeMenu() {
    return toggleMenu(false);
  };

  var handleOnSelect = function handleOnSelect(val, label) {
    setLabel(label);
    onSelect(val);
  };

  var randomId = "id-" + (Math.random() * 10).toFixed(3);
  var showText = errorText ? errorText : helperText ? helperText : "";
  var isError = !!errorText;
  return _react["default"].createElement(_OnClickOutside["default"], {
    onClickOutside: closeMenu
  }, _react["default"].createElement(DropdownWrapper, {
    width: width,
    onClick: toggle,
    tabIndex: "0",
    "aria-describedby": "dropdown-help",
    "aria-activedescendant": randomId + "-selected"
  }, _react["default"].createElement(HeaderWrapper, {
    isMenuOpen: isMenuOpen,
    isError: isError
  }, _react["default"].createElement(LeftSection, null, !!props.leftIcon && _react["default"].createElement(_Icon["default"], {
    name: leftIcon,
    size: 16,
    color: _color.grey.z800
  }), _react["default"].createElement(Span, null, label)), _react["default"].createElement(RightSection, {
    isMenuOpen: isMenuOpen
  }, _react["default"].createElement(_Icon["default"], {
    name: rightIcon,
    size: 14,
    color: _color.grey.z400
  }))), !!showText && _react["default"].createElement(HelperText, {
    isError: isError,
    id: "dropdown-help"
  }, showText), _react["default"].createElement(MenuHolder, {
    showArrow: showArrow,
    isOpen: isMenuOpen
  }, isMenuOpen && _react["default"].createElement(_react["default"].Fragment, null, showArrow && _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement(MenuHolder.Arrow, null), _react["default"].createElement(Cover, null)), _react["default"].createElement(MenuWrapper, {
    tabIndex: "-1"
  }, (0, _commonHelper.cloneChildren)(children, {
    _passClick: handleOnSelect,
    id: randomId
  }))))));
};

var MenuHolder = _styledComponents["default"].span.withConfig({
  displayName: "Dropdown__MenuHolder",
  componentId: "sc-9al8nu-0"
})(["position:absolute;top:", ";width:100%;border-radius:", ";background:", ";box-shadow:0px 1px 8px rgba(28,28,28,0.15);z-index:10;opacity:", ";transition:all ", " ease;"], function (props) {
  return props.showArrow ? "6.8rem" : "6.3rem";
}, _misc["default"].radius, _color.white, function (props) {
  return props.isOpen ? "1" : "0";
}, _misc["default"].animationDuration);

var MenuWrapper = _styledComponents["default"].section.withConfig({
  displayName: "Dropdown__MenuWrapper",
  componentId: "sc-9al8nu-1"
})(["margin:1rem 0;"]);

var Cover = _styledComponents["default"].span.withConfig({
  displayName: "Dropdown__Cover",
  componentId: "sc-9al8nu-2"
})(["position:absolute;right:1.5rem;width:2.5rem;height:1rem;top:0;background:white;"]);

MenuHolder.Arrow = _styledComponents["default"].div.withConfig({
  displayName: "Dropdown__Arrow",
  componentId: "sc-9al8nu-3"
})(["position:absolute;width:1.4rem;height:1.4rem;transform:rotate(45deg);top:-7px;right:2rem;background:white;box-shadow:0px -1px 3px rgba(28,28,28,0.15);"]);

var RightSection = _styledComponents["default"].span.withConfig({
  displayName: "Dropdown__RightSection",
  componentId: "sc-9al8nu-4"
})(["transform:", ";transition:transform ", ";"], function (props) {
  return props.isMenuOpen ? "rotate(180deg)" : "none";
}, _misc["default"].animationDuration);

var HelperText = _styledComponents["default"].p.withConfig({
  displayName: "Dropdown__HelperText",
  componentId: "sc-9al8nu-5"
})(["font-size:11px;margin:8px 0;color:", ";"], function (props) {
  return props.isError ? _color.red.z400 : _color.grey.z400;
});

Dropdown.propTypes = {
  value: _propTypes["default"].string,
  width: _propTypes["default"].string,
  leftIcon: _propTypes["default"].string,
  rightIcon: _propTypes["default"].string,
  children: _propTypes["default"].node,
  onSelect: _propTypes["default"].func.isRequired,
  helperText: _propTypes["default"].string,
  errorText: _propTypes["default"].string,
  showArrow: _propTypes["default"].bool
};
Dropdown.defaultProps = {
  label: "Choose...",
  value: "",
  width: "",
  leftIcon: "",
  rightIcon: "downTriangle",
  children: null,
  showArrow: false
};

var Span = _styledComponents["default"].span.withConfig({
  displayName: "Dropdown__Span",
  componentId: "sc-9al8nu-6"
})(["margin-left:1rem;margin-top:0.2rem;color:", ";font-size:1.8rem;font-weight:300;overflow:hidden;text-overflow:ellipsis;"], _color.grey.z800);

var LeftSection = _styledComponents["default"].div.withConfig({
  displayName: "Dropdown__LeftSection",
  componentId: "sc-9al8nu-7"
})(["display:flex;align-items:center;margin-right:20px;overflow:hidden;white-space:nowrap;"]);

var HeaderWrapper = _styledComponents["default"].div.withConfig({
  displayName: "Dropdown__HeaderWrapper",
  componentId: "sc-9al8nu-8"
})(["display:flex;align-items:center;justify-content:space-between;padding:", ";font-size:18px;border:", ";border-radius:", ";cursor:pointer;background:", ";"], function (props) {
  return getPadding(props);
}, function (props) {
  return getBorder(props);
}, _misc["default"].radius, function (props) {
  return props.isError ? _color.red.z100 : "#fff";
});

var getPadding = function getPadding(props) {
  if (props.isError) {
    return "15px 18.5px";
  } else {
    if (props.isMenuOpen) {
      return "15.5px 19px";
    } else {
      return "16.5px 20px";
    }
  }
};

var getBorder = function getBorder(props) {
  return function () {
    if (props.isError) {
      return "2px solid " + _color.red.z600;
    } else {
      if (props.isMenuOpen) {
        return "1.5px solid " + _color.green.z500;
      } else {
        return "0.5px solid " + _color.grey.z300;
      }
    }
  };
};

var DropdownWrapper = _styledComponents["default"].div.withConfig({
  displayName: "Dropdown__DropdownWrapper",
  componentId: "sc-9al8nu-9"
})(["position:relative;width:", ";"], function (props) {
  return props.width ? props.width : "max-content";
});

var _default = Dropdown;
exports["default"] = _default;
