// webuikit/esm/RestaurantOrderCard/orderCardHelpers.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getState = exports.getO2Info = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _constants = require("./constants");

var _color = require("../tokens/color");

var _disabledState, _activeState;

var disabledState = (_disabledState = {}, (0, _defineProperty2["default"])(_disabledState, _constants.IMG_FILTER, "grayscale(100%)"), (0, _defineProperty2["default"])(_disabledState, _constants.HEADING_COLOR, _color.grey.z500), (0, _defineProperty2["default"])(_disabledState, _constants.TYPE_COLOR, _color.grey.z500), (0, _defineProperty2["default"])(_disabledState, _constants.RATING_COLOR, _color.grey.z300), _disabledState);
var activeState = (_activeState = {}, (0, _defineProperty2["default"])(_activeState, _constants.IMG_FILTER, "none"), (0, _defineProperty2["default"])(_activeState, _constants.HEADING_COLOR, _color.black), (0, _defineProperty2["default"])(_activeState, _constants.TYPE_COLOR, _color.grey.z800), (0, _defineProperty2["default"])(_activeState, _constants.RATING_COLOR, _color.green.z500), _activeState);

var getOfferCodePadding = function getOfferCodePadding(_ref) {
  var codeStatement = _ref.codeStatement;
  return codeStatement ? "1rem" : "0";
};

var DType = _styledComponents["default"].span.withConfig({
  displayName: "orderCardHelpers__DType",
  componentId: "sc-1h09ln4-0"
})(["color:", ";padding-top:0.6rem;font-size:1.6rem;line-height:2.3rem;padding-left:", ";"], _color.red.z600, getOfferCodePadding);

var getO2Info = function getO2Info(props) {
  if (props.disabledStatement) {
    return _react["default"].createElement("section", null, _react["default"].createElement(DType, null, props.disabledStatement));
  }

  if (props.discount || props.codeStatement) {
    return _react["default"].createElement("section", null, _react["default"].createElement(DType, null, props.discount), _react["default"].createElement(DType, {
      codeStatement: true
    }, props.codeStatement));
  }
};

exports.getO2Info = getO2Info;

var getState = function getState(key) {
  return function (_ref2) {
    var disabled = _ref2.disabled;
    return disabled ? disabledState[key] : activeState[key];
  };
};

exports.getState = getState;
