// webuikit/ImageUploader/ImageUploader.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _lodashEs = require("lodash-es");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _componentHelpers = require("./componentHelpers");

var _misc = _interopRequireDefault(require("../tokens/misc"));

// Libraries
// Helpers
var ImageUploader = function ImageUploader(props) {
  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      hover = _useState2[0],
      setHover = _useState2[1];

  var _useState3 = (0, _react.useState)([]),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      urlMap = _useState4[0],
      setUrlMap = _useState4[1];

  var _useState5 = (0, _react.useState)(["File is too large"]),
      _useState6 = (0, _slicedToArray2["default"])(_useState5, 2),
      errMap = _useState6[0],
      setErrMap = _useState6[1];

  (0, _react.useEffect)(function () {
    var params = {
      onChange: props.onChange,
      setHover: setHover,
      setUrlMap: setUrlMap,
      setErrMap: setErrMap,
      urlMap: urlMap,
      errMap: errMap
    }; // creating events on mount

    (0, _componentHelpers.eventManager)(window.addEventListener, params);
    return function () {
      // removing events on unmount
      (0, _componentHelpers.eventManager)(window.removeEventListener, params);
    };
  }, [errMap, props.onChange, urlMap]);
  var params = (0, _objectSpread2["default"])({}, props, {
    setHover: setHover,
    hover: hover,
    urlMap: urlMap,
    errMap: errMap,
    setUrlMap: setUrlMap,
    setErrMap: setErrMap
  });
  return _react["default"].createElement(ImageUploader.Element, {
    className: props.classname
  }, _react["default"].createElement("header", null, _react["default"].createElement("label", null, props.label), props.secondSourceLink ? _react["default"].createElement("a", {
    className: "secondary-source",
    href: props.secondSourceLink
  }, props.secondSourceLbl) : null), (0, _componentHelpers.viewRenderer)(params));
};

ImageUploader.Element = _styledComponents["default"].section.withConfig({
  displayName: "ImageUploader__Element",
  componentId: "twihiv-0"
})(["box-shadow:0 1px 8px rgba(28,28,28,0.15);border-radius:", ";padding:20px;header{padding:4px 0 24px;display:flex;align-items:center;justify-content:space-between;}label{font-size:17px;line-height:20px;}.secondary-source{cursor:pointer;font-size:14px;line-height:20px;color:#28a745;text-decoration:none;}"], _misc["default"].radius);
ImageUploader.defaultProps = {
  onChange: _lodashEs.noop,
  multiple: false
};
ImageUploader.propTypes = {
  label: _propTypes["default"].string.isRequired,
  btnLabel: _propTypes["default"].string.isRequired,
  secondSourceLink: _propTypes["default"].string,
  secondSourceLbl: _propTypes["default"].string,
  classname: _propTypes["default"].string,
  btnSubTitle: _propTypes["default"].string,
  onChange: _propTypes["default"].func,
  multiple: _propTypes["default"].bool
};
var _default = ImageUploader;
exports["default"] = _default;
