// webuikit/esm/Radio/Radio.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Radio = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _lodashEs = require("lodash-es");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Label = _interopRequireDefault(require("../Label"));

var _color = require("../tokens/color");

var _spacing = require("../tokens/spacing");

var _misc = _interopRequireDefault(require("../tokens/misc"));

var _radioHelpers = require("./radioHelpers");

// libraries
// helpers
var Radio = function Radio(props) {
  return _react["default"].createElement(Radio.Wrapper, null, _react["default"].createElement(_Label["default"], null, _react["default"].createElement(Radio.Element, (0, _extends2["default"])({}, props, {
    type: "radio"
  })), _react["default"].createElement(Radio.Label, null, props.label)));
};

exports.Radio = Radio;
Radio.Label = _styledComponents["default"].span.withConfig({
  displayName: "Radio__Label",
  componentId: "iunt8g-0"
})(["padding-top:0.2rem;"]);
Radio.Wrapper = _styledComponents["default"].section.withConfig({
  displayName: "Radio__Wrapper",
  componentId: "iunt8g-1"
})(["width:max-content;margin-bottom:", ";margin-right:", ";"], _spacing.zSpacing.z3, _spacing.zSpacing.z4);
Radio.Element = _styledComponents["default"].input.withConfig({
  displayName: "Radio__Element",
  componentId: "iunt8g-2"
})(["appearance:none;cursor:", ";outline:0;border-radius:50%;margin-right:0.8rem;width:1.8rem;height:1.8rem;background:", ";display:inline-flex;border:0.1rem solid ", ";box-sizing:border-box;position:relative;justify-content:center;align-items:center;transition:all ", " linear;", ":hover &{border-color:", ";}&:checked::before{position:absolute;width:1.1rem;height:1.1rem;left:50%;top:50%;transform:translate(-50%,-50%);content:\"\";border-radius:50%;background-color:", ";transition:all ", " linear;}"], (0, _radioHelpers.getAttributes)("cursor"), _color.white, (0, _radioHelpers.getAttributes)("borderColor"), _misc["default"].animationDuration, Radio.Wrapper, (0, _radioHelpers.getAttributes)("hoverBorderColor"), (0, _radioHelpers.getAttributes)("selectedBgColor"), _misc["default"].animationDuration);
Radio.propTypes = {
  "class": _propTypes["default"].string,
  label: _propTypes["default"].string,
  value: _propTypes["default"].string,
  color: _propTypes["default"].string,
  checked: _propTypes["default"].bool,
  onClick: _propTypes["default"].func,
  disabled: _propTypes["default"].bool
};
Radio.defaultProps = {
  value: "1",
  disabled: false,
  onClick: _lodashEs.noop,
  checked: false
};
var _default = Radio;
exports["default"] = _default;
