// webuikit/SimpleCard/SimpleCard.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _Button = _interopRequireDefault(require("../Button"));

var _color = require("../tokens/color");

var _Typography = require("../Typography");

var _PageLayout = require("../Layouts/PageLayout");

var _Input = _interopRequireDefault(require("../InputTypes/Input"));

//../tokens/color
var ZImage = _styledComponents["default"].img.withConfig({
  displayName: "SimpleCard__ZImage",
  componentId: "sc-921v1u-0"
})(["max-width:100%;background-color:grey;& > *{margin-bottom:1rem;}"]);

var SimpleCard = function SimpleCard(props) {
  return _react["default"].createElement(_PageLayout.StackedSectionLeft, null, _react["default"].createElement(ZImage, {
    src: "https://images.pexels.com/photos/376464/pexels-photo-376464.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
    maxHeight: "384px",
    width: "100%"
  }), _react["default"].createElement(_Typography.Tag, null, "Tags Text"), _react["default"].createElement(_Typography.H1, null, "Card Headline text long headlines"), _react["default"].createElement(_Typography.P, {
    size: "2.4rem"
  }, "I have hinted that I would often jerk poor Queequeg from between the whale and the ship\u2014where he would occasionally fall, from the incessant.", _react["default"].createElement(_Button["default"], {
    btnColor: "red",
    size: "large",
    appearance: "link",
    icon: "right-triangle",
    iconAlign: "right",
    fontSize: "24px"
  }, "Read More")), _react["default"].createElement(_Button["default"], {
    btnColor: "red"
  }, "Normal"));
};

var _default = SimpleCard;
exports["default"] = _default;
